{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"Andrej_QIIME2_PE_ENA (4).ipynb","provenance":[],"collapsed_sections":[]},"kernelspec":{"display_name":"Python 3","name":"python3"}},"cells":[{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"lL7p4sYdDKTj","executionInfo":{"status":"ok","timestamp":1646125450305,"user_tz":0,"elapsed":18445,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"875fa119-6718-4cbd-c5bb-af7e2b2c4c1f"},"source":["from google.colab import drive\n","drive.mount('/content/drive')"],"execution_count":1,"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}]},{"cell_type":"code","metadata":{"id":"r0-m09lju6Xl","executionInfo":{"status":"ok","timestamp":1646125460496,"user_tz":0,"elapsed":3246,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}}},"source":["!cp /content/drive/MyDrive/AF/*.qz* /content/sample_data/\n"],"execution_count":2,"outputs":[]},{"cell_type":"code","metadata":{"id":"TBprUehrsemF","executionInfo":{"status":"ok","timestamp":1646125461935,"user_tz":0,"elapsed":531,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}}},"source":["!cp /content/drive/MyDrive/AF/*.txt /content/sample_data/"],"execution_count":3,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"4rETZPSq1uOA"},"source":["#**Skip these parts** Start at step 1."]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"PqFj1j902k_E","executionInfo":{"status":"ok","timestamp":1646125478565,"user_tz":0,"elapsed":197,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"6a0f71d9-ac88-4da4-a057-3c2e29682bec"},"source":["cd /content/sample_data/"],"execution_count":5,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/sample_data\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"zfIRiYj2wX-l","executionInfo":{"status":"ok","timestamp":1646125484279,"user_tz":0,"elapsed":3348,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"48d46148-8ab9-49e6-f6a2-3c7ca284d90c"},"source":["!gsutil cp demux-filtered.qza gs://q3_2021/"],"execution_count":6,"outputs":[{"output_type":"stream","name":"stdout","text":["CommandException: No URLs matched: demux-filtered.qza\n"]}]},{"cell_type":"code","metadata":{"id":"wu6Kou1lZoWQ"},"source":["from google.colab import auth\n","auth.authenticate_user()"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"oGeJEzPaV-Wt","executionInfo":{"status":"ok","timestamp":1630552342874,"user_tz":-60,"elapsed":689,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"05885788-92df-4210-d3ab-3b5ba238a193"},"source":["!wget \"https://github.com/enasequence/enaBrowserTools/archive/v1.6.tar.gz\""],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["--2021-09-02 03:12:22--  https://github.com/enasequence/enaBrowserTools/archive/v1.6.tar.gz\n","Resolving github.com (github.com)... 192.30.255.113\n","Connecting to github.com (github.com)|192.30.255.113|:443... connected.\n","HTTP request sent, awaiting response... 302 Found\n","Location: https://codeload.github.com/enasequence/enaBrowserTools/tar.gz/v1.6 [following]\n","--2021-09-02 03:12:22--  https://codeload.github.com/enasequence/enaBrowserTools/tar.gz/v1.6\n","Resolving codeload.github.com (codeload.github.com)... 192.30.255.121\n","Connecting to codeload.github.com (codeload.github.com)|192.30.255.121|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: unspecified [application/x-gzip]\n","Saving to: ‘v1.6.tar.gz.1’\n","\n","v1.6.tar.gz.1           [ <=>                ]  19.12K  --.-KB/s    in 0.007s  \n","\n","2021-09-02 03:12:22 (2.49 MB/s) - ‘v1.6.tar.gz.1’ saved [19577]\n","\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"YD-EHbJ8WHNL","executionInfo":{"status":"ok","timestamp":1630552396907,"user_tz":-60,"elapsed":234,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"b9877e9a-7cc1-4739-8f4d-5e6bf08d0b35"},"source":["!tar -xvf /content/v1.6.tar.gz.1"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["tar: /content/v1.6.tar.gz.1: Cannot open: No such file or directory\n","tar: Error is not recoverable: exiting now\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"HwyOw1coWKJn","executionInfo":{"status":"ok","timestamp":1630552260348,"user_tz":-60,"elapsed":241,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"08d00808-d120-4452-da59-b3a1bd766427"},"source":["cd /content/enaBrowserTools-1.6/python3"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["[Errno 2] No such file or directory: '/content/enaBrowserTools-1.6/python3'\n","/content/sample_data\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Sz811U31kQ6a","executionInfo":{"status":"ok","timestamp":1630552309540,"user_tz":-60,"elapsed":268,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"c75339ba-75bc-4a07-9b00-e9ff79b3c09d"},"source":["!./FFA PRJEB27564"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["/bin/bash: ./FFA: Permission denied\n"]}]},{"cell_type":"code","metadata":{"id":"MCvGSaJq1_0m"},"source":["import os\n","import shutil\n","current_wkd=\"/content/enaBrowserTools-1.6/python3/\"\n","dir_dst= \"/content/sample_data\"\n","for root, dir, files in os.walk(top=current_wkd):\n","    for file in files:\n","        if file.endswith(\".gz\"):  #match files that match this extension\n","            shutil.move(os.path.join(root, file), dir_dst)"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"Q5DSLP7C_w1v"},"source":["#**1. Set up QIIME2, Download Qiime2, Miniconda and create a QIIME2 environment**"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"iuEwG2Qs1STx","executionInfo":{"status":"ok","timestamp":1646125489497,"user_tz":0,"elapsed":894,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"2eac225d-dcdd-4930-8613-e2d2e008169b"},"source":["!wget https://data.qiime2.org/distro/core/qiime2-2020.11-py36-linux-conda.yml"],"execution_count":7,"outputs":[{"output_type":"stream","name":"stdout","text":["--2022-03-01 09:04:48--  https://data.qiime2.org/distro/core/qiime2-2020.11-py36-linux-conda.yml\n","Resolving data.qiime2.org (data.qiime2.org)... 54.200.1.12\n","Connecting to data.qiime2.org (data.qiime2.org)|54.200.1.12|:443... connected.\n","HTTP request sent, awaiting response... 302 FOUND\n","Location: https://raw.githubusercontent.com/qiime2/environment-files/master/2020.11/release/qiime2-2020.11-py36-linux-conda.yml [following]\n","--2022-03-01 09:04:49--  https://raw.githubusercontent.com/qiime2/environment-files/master/2020.11/release/qiime2-2020.11-py36-linux-conda.yml\n","Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.110.133, 185.199.111.133, 185.199.109.133, ...\n","Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.110.133|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 9474 (9.3K) [text/plain]\n","Saving to: ‘qiime2-2020.11-py36-linux-conda.yml’\n","\n","qiime2-2020.11-py36 100%[===================>]   9.25K  --.-KB/s    in 0s      \n","\n","2022-03-01 09:04:49 (86.8 MB/s) - ‘qiime2-2020.11-py36-linux-conda.yml’ saved [9474/9474]\n","\n"]}]},{"cell_type":"markdown","metadata":{"id":"uPTF7ZE_NZNx"},"source":["#**Download miniconda**"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"CeZ_7bXI1VIN","executionInfo":{"status":"ok","timestamp":1646125515291,"user_tz":0,"elapsed":22937,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"18987abd-d01a-4e24-e5d9-320accdeee90"},"source":["!wget -c https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n","!chmod +x Miniconda3-latest-Linux-x86_64.sh\n","!bash ./Miniconda3-latest-Linux-x86_64.sh -b -f -p /usr/local\n","# can change to another channel or package\n","#!conda install -q -y --prefix /usr/local ujson\n","import sys\n","sys.path.append(\"/usr/local/conda/lib/python3.6/site-packages/\")\n"],"execution_count":8,"outputs":[{"output_type":"stream","name":"stdout","text":["--2022-03-01 09:04:52--  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n","Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.130.3, 104.16.131.3, 2606:4700::6810:8303, ...\n","Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.130.3|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 75660608 (72M) [application/x-sh]\n","Saving to: ‘Miniconda3-latest-Linux-x86_64.sh’\n","\n","Miniconda3-latest-L 100%[===================>]  72.16M   195MB/s    in 0.4s    \n","\n","2022-03-01 09:04:52 (195 MB/s) - ‘Miniconda3-latest-Linux-x86_64.sh’ saved [75660608/75660608]\n","\n","PREFIX=/usr/local\n","Unpacking payload ...\n","Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\bdone\n","Solving environment: - \b\b\\ \b\b| \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /usr/local\n","\n","  added / updated specs:\n","    - _libgcc_mutex==0.1=main\n","    - _openmp_mutex==4.5=1_gnu\n","    - brotlipy==0.7.0=py39h27cfd23_1003\n","    - ca-certificates==2021.10.26=h06a4308_2\n","    - certifi==2021.10.8=py39h06a4308_2\n","    - cffi==1.15.0=py39hd667e15_1\n","    - charset-normalizer==2.0.4=pyhd3eb1b0_0\n","    - conda-content-trust==0.1.1=pyhd3eb1b0_0\n","    - conda-package-handling==1.7.3=py39h27cfd23_1\n","    - conda==4.11.0=py39h06a4308_0\n","    - cryptography==36.0.0=py39h9ce1e76_0\n","    - idna==3.3=pyhd3eb1b0_0\n","    - ld_impl_linux-64==2.35.1=h7274673_9\n","    - libffi==3.3=he6710b0_2\n","    - libgcc-ng==9.3.0=h5101ec6_17\n","    - libgomp==9.3.0=h5101ec6_17\n","    - libstdcxx-ng==9.3.0=hd4cf53a_17\n","    - ncurses==6.3=h7f8727e_2\n","    - openssl==1.1.1m=h7f8727e_0\n","    - pip==21.2.4=py39h06a4308_0\n","    - pycosat==0.6.3=py39h27cfd23_0\n","    - pycparser==2.21=pyhd3eb1b0_0\n","    - pyopenssl==21.0.0=pyhd3eb1b0_1\n","    - pysocks==1.7.1=py39h06a4308_0\n","    - python==3.9.7=h12debd9_1\n","    - readline==8.1.2=h7f8727e_1\n","    - requests==2.27.1=pyhd3eb1b0_0\n","    - ruamel_yaml==0.15.100=py39h27cfd23_0\n","    - setuptools==58.0.4=py39h06a4308_0\n","    - six==1.16.0=pyhd3eb1b0_0\n","    - sqlite==3.37.0=hc218d9a_0\n","    - tk==8.6.11=h1ccaba5_0\n","    - tqdm==4.62.3=pyhd3eb1b0_1\n","    - tzdata==2021e=hda174b7_0\n","    - urllib3==1.26.7=pyhd3eb1b0_0\n","    - wheel==0.37.1=pyhd3eb1b0_0\n","    - xz==5.2.5=h7b6447c_0\n","    - yaml==0.2.5=h7b6447c_0\n","    - zlib==1.2.11=h7f8727e_4\n","\n","\n","The following NEW packages will be INSTALLED:\n","\n","  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main\n","  _openmp_mutex      pkgs/main/linux-64::_openmp_mutex-4.5-1_gnu\n","  brotlipy           pkgs/main/linux-64::brotlipy-0.7.0-py39h27cfd23_1003\n","  ca-certificates    pkgs/main/linux-64::ca-certificates-2021.10.26-h06a4308_2\n","  certifi            pkgs/main/linux-64::certifi-2021.10.8-py39h06a4308_2\n","  cffi               pkgs/main/linux-64::cffi-1.15.0-py39hd667e15_1\n","  charset-normalizer pkgs/main/noarch::charset-normalizer-2.0.4-pyhd3eb1b0_0\n","  conda              pkgs/main/linux-64::conda-4.11.0-py39h06a4308_0\n","  conda-content-tru~ pkgs/main/noarch::conda-content-trust-0.1.1-pyhd3eb1b0_0\n","  conda-package-han~ pkgs/main/linux-64::conda-package-handling-1.7.3-py39h27cfd23_1\n","  cryptography       pkgs/main/linux-64::cryptography-36.0.0-py39h9ce1e76_0\n","  idna               pkgs/main/noarch::idna-3.3-pyhd3eb1b0_0\n","  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.35.1-h7274673_9\n","  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2\n","  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.3.0-h5101ec6_17\n","  libgomp            pkgs/main/linux-64::libgomp-9.3.0-h5101ec6_17\n","  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.3.0-hd4cf53a_17\n","  ncurses            pkgs/main/linux-64::ncurses-6.3-h7f8727e_2\n","  openssl            pkgs/main/linux-64::openssl-1.1.1m-h7f8727e_0\n","  pip                pkgs/main/linux-64::pip-21.2.4-py39h06a4308_0\n","  pycosat            pkgs/main/linux-64::pycosat-0.6.3-py39h27cfd23_0\n","  pycparser          pkgs/main/noarch::pycparser-2.21-pyhd3eb1b0_0\n","  pyopenssl          pkgs/main/noarch::pyopenssl-21.0.0-pyhd3eb1b0_1\n","  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py39h06a4308_0\n","  python             pkgs/main/linux-64::python-3.9.7-h12debd9_1\n","  readline           pkgs/main/linux-64::readline-8.1.2-h7f8727e_1\n","  requests           pkgs/main/noarch::requests-2.27.1-pyhd3eb1b0_0\n","  ruamel_yaml        pkgs/main/linux-64::ruamel_yaml-0.15.100-py39h27cfd23_0\n","  setuptools         pkgs/main/linux-64::setuptools-58.0.4-py39h06a4308_0\n","  six                pkgs/main/noarch::six-1.16.0-pyhd3eb1b0_0\n","  sqlite             pkgs/main/linux-64::sqlite-3.37.0-hc218d9a_0\n","  tk                 pkgs/main/linux-64::tk-8.6.11-h1ccaba5_0\n","  tqdm               pkgs/main/noarch::tqdm-4.62.3-pyhd3eb1b0_1\n","  tzdata             pkgs/main/noarch::tzdata-2021e-hda174b7_0\n","  urllib3            pkgs/main/noarch::urllib3-1.26.7-pyhd3eb1b0_0\n","  wheel              pkgs/main/noarch::wheel-0.37.1-pyhd3eb1b0_0\n","  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0\n","  yaml               pkgs/main/linux-64::yaml-0.2.5-h7b6447c_0\n","  zlib               pkgs/main/linux-64::zlib-1.2.11-h7f8727e_4\n","\n","\n","Preparing transaction: - \b\b\\ \b\b| \b\b/ \b\bdone\n","Executing transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","installation finished.\n","WARNING:\n","    You currently have a PYTHONPATH environment variable set. This may cause\n","    unexpected behavior when running the Python interpreter in Miniconda3.\n","    For best results, please verify that your PYTHONPATH only points to\n","    directories of packages that are compatible with the Python interpreter\n","    in Miniconda3: /usr/local\n"]}]},{"cell_type":"markdown","metadata":{"id":"fmQeQT0zADE6"},"source":["#**Create a conda enviroment for Qiime2**"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"o22YcB-S2AGx","executionInfo":{"status":"ok","timestamp":1646125933371,"user_tz":0,"elapsed":375727,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"34532682-4166-4da6-bc85-3b7bcec812b2"},"source":["!conda env create -n qiime2-2020.11 --file qiime2-2020.11-py36-linux-conda.yml"],"execution_count":9,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","\n","Downloading and Extracting Packages\n","q2-longitudinal-2020 | 7.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.55s/it]\n","curl-7.71.1          | 139 KB    | : 100% 1.0/1 [00:00<00:00,  3.06it/s]              \n","libgcc-devel_linux-6 | 3.6 MB    | : 100% 1.0/1 [00:00<00:00,  1.55it/s]\n","sepp-4.3.10          | 6.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.41s/it]\n","r-praise-1.0.0       | 24 KB     | : 100% 1.0/1 [00:00<00:00, 23.38it/s]\n","notebook-6.1.5       | 6.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.84s/it]\n","r-assertthat-0.2.1   | 70 KB     | : 100% 1.0/1 [00:00<00:00, 19.38it/s]\n","gxx_impl_linux-64-7. | 9.5 MB    | : 100% 1.0/1 [00:01<00:00,  1.88s/it]               \n","pandocfilters-1.4.2  | 9 KB      | : 100% 1.0/1 [00:00<00:00, 22.02it/s]\n","q2-emperor-2020.11.1 | 14 KB     | : 100% 1.0/1 [00:00<00:00,  3.92it/s]\n","q2-alignment-2020.11 | 19 KB     | : 100% 1.0/1 [00:00<00:00,  3.71it/s]\n","olefile-0.46         | 32 KB     | : 100% 1.0/1 [00:00<00:00, 21.93it/s]\n","bioconductor-summari | 2.6 MB    | : 100% 1.0/1 [00:00<00:00,  2.30it/s]\n","perl-data-dumper-2.1 | 35 KB     | : 100% 1.0/1 [00:00<00:00, 26.13it/s]\n","idna-2.10            | 52 KB     | : 100% 1.0/1 [00:00<00:00, 19.59it/s]\n","bioconductor-genomei | 4.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.01it/s]\n","entrez-direct-13.9   | 5.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.30it/s]\n","sed-4.8              | 264 KB    | : 100% 1.0/1 [00:00<00:00, 11.04it/s]\n","scipy-1.5.3          | 18.6 MB   | : 100% 1.0/1 [00:03<00:00,  3.71s/it]\n","libwebp-base-1.1.0   | 864 KB    | : 100% 1.0/1 [00:00<00:00,  5.07it/s]\n","xorg-libxfixes-5.0.3 | 17 KB     | : 100% 1.0/1 [00:00<00:00, 23.67it/s]\n","gettext-0.19.8.1     | 3.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.14s/it]\n","typing_extensions-3. | 25 KB     | : 100% 1.0/1 [00:00<00:00, 26.22it/s]\n","kernel-headers_linux | 707 KB    | : 100% 1.0/1 [00:00<00:00,  3.72it/s]\n","r-processx-3.4.5     | 296 KB    | : 100% 1.0/1 [00:00<00:00, 10.87it/s]\n","libgomp-9.3.0        | 376 KB    | : 100% 1.0/1 [00:00<00:00, 10.12it/s]\n","r-rematch2-2.1.2     | 52 KB     | : 100% 1.0/1 [00:00<00:00, 21.92it/s]\n","perl-http-daemon-6.0 | 14 KB     | : 100% 1.0/1 [00:00<00:00, 28.76it/s]\n","xz-5.2.5             | 343 KB    | : 100% 1.0/1 [00:00<00:00,  9.12it/s]\n","q2-phylogeny-2020.11 | 46 KB     | : 100% 1.0/1 [00:00<00:00,  3.68it/s]\n","r-ps-1.5.0           | 290 KB    | : 100% 1.0/1 [00:00<00:00,  2.55it/s]                \n","python-dateutil-2.8. | 220 KB    | : 100% 1.0/1 [00:00<00:00, 14.11it/s]\n","bioconductor-xvector | 733 KB    | : 100% 1.0/1 [00:00<00:00,  7.14it/s]\n","liblapacke-3.8.0     | 11 KB     | : 100% 1.0/1 [00:00<00:00, 25.31it/s]\n","libssh2-1.9.0        | 225 KB    | : 100% 1.0/1 [00:00<00:00,  8.23it/s]\n","six-1.15.0           | 14 KB     | : 100% 1.0/1 [00:00<00:00, 23.87it/s]\n","pexpect-4.8.0        | 47 KB     | : 100% 1.0/1 [00:00<00:00, 22.41it/s]\n","perl-xml-parser-2.44 | 165 KB    | : 100% 1.0/1 [00:00<00:00,  2.38it/s]                \n","tktable-2.10         | 89 KB     | : 100% 1.0/1 [00:00<00:00, 19.84it/s]\n","bzip2-1.0.8          | 484 KB    | : 100% 1.0/1 [00:00<00:00,  7.43it/s]\n","xorg-libxtst-1.2.3   | 31 KB     | : 100% 1.0/1 [00:00<00:00, 19.55it/s]\n","perl-libwww-perl-6.3 | 94 KB     | : 100% 1.0/1 [00:00<00:00, 17.45it/s]\n","fribidi-1.0.10       | 112 KB    | : 100% 1.0/1 [00:00<00:00, 13.82it/s]\n","bokeh-2.2.3          | 7.0 MB    | : 100% 1.0/1 [00:02<00:00,  2.45s/it]\n","perl-pathtools-3.75  | 41 KB     | : 100% 1.0/1 [00:00<00:00, 23.22it/s]\n","r-ggplot2-3.3.2      | 3.9 MB    | : 100% 1.0/1 [00:00<00:00,  1.85it/s]\n","q2-types-2020.11.1   | 111 KB    | : 100% 1.0/1 [00:00<00:00,  2.88it/s]\n","cryptography-3.3.1   | 624 KB    | : 100% 1.0/1 [00:00<00:00,  1.28it/s]\n","xorg-libxrender-0.9. | 31 KB     | : 100% 1.0/1 [00:00<00:00, 25.82it/s]\n","ipython-7.16.1       | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.53it/s]\n","send2trash-1.5.0     | 12 KB     | : 100% 1.0/1 [00:00<00:00, 30.14it/s]\n","bioconductor-biocgen | 699 KB    | : 100% 1.0/1 [00:00<00:00,  7.10it/s]\n","q2-composition-2020. | 169 KB    | : 100% 1.0/1 [00:00<00:00,  2.83it/s]\n","r-png-0.1_7          | 58 KB     | : 100% 1.0/1 [00:00<00:00, 21.11it/s]\n","pysocks-1.7.1        | 27 KB     | : 100% 1.0/1 [00:00<00:00, 26.94it/s]\n","python-3.6.12        | 38.4 MB   | : 100% 1.0/1 [00:06<00:00,  6.59s/it]               \n","bioconductor-rhtslib | 1.8 MB    | : 100% 1.0/1 [00:00<00:00,  2.17it/s]\n","binutils_linux-64-2. | 22 KB     | : 100% 1.0/1 [00:00<00:00, 23.86it/s]\n","pytest-6.1.2         | 418 KB    | : 100% 1.0/1 [00:00<00:00,  5.67it/s]\n","r-rcolorbrewer-1.1_2 | 64 KB     | : 100% 1.0/1 [00:00<00:00, 24.29it/s]\n","bioconductor-genomic | 2.3 MB    | : 100% 1.0/1 [00:00<00:00,  2.77it/s]\n","r-diffobj-0.3.2      | 1.0 MB    | : 100% 1.0/1 [00:00<00:00,  2.07it/s]\n","perl-digest-md5-2.55 | 18 KB     | : 100% 1.0/1 [00:00<00:00, 30.32it/s]\n","r-zeallot-0.1.0      | 63 KB     | : 100% 1.0/1 [00:00<00:00, 23.38it/s]\n","_r-mutex-1.0.1       | 3 KB      | : 100% 1.0/1 [00:00<00:00, 27.04it/s]\n","alsa-lib-1.2.3       | 560 KB    | : 100% 1.0/1 [00:00<00:00,  7.03it/s]\n","bioconductor-delayed | 2.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.26it/s]\n","libdeflate-1.6       | 60 KB     | : 100% 1.0/1 [00:00<00:00, 20.35it/s]\n","nest-asyncio-1.4.3   | 9 KB      | : 100% 1.0/1 [00:00<00:00, 16.65it/s]\n","r-stringr-1.4.0      | 210 KB    | : 100% 1.0/1 [00:00<00:00, 16.58it/s]\n","bioconductor-shortre | 5.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.28it/s]\n","perl-mime-base64-3.1 | 15 KB     | : 100% 1.0/1 [00:00<00:00, 29.95it/s]\n","r-rprojroot-2.0.2    | 110 KB    | : 100% 1.0/1 [00:00<00:00, 20.09it/s]\n","terminado-0.9.1      | 25 KB     | : 100% 1.0/1 [00:00<00:00, 18.26it/s]\n","perl-constant-1.33   | 10 KB     | : 100% 1.0/1 [00:00<00:00, 27.01it/s]\n","cycler-0.10.0        | 9 KB      | : 100% 1.0/1 [00:00<00:00, 26.78it/s]\n","pandas-1.1.5         | 11.3 MB   | : 100% 1.0/1 [00:03<00:00,  3.17s/it]               \n","q2-cutadapt-2020.11. | 6.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.43s/it]\n","perl-app-cpanminus-1 | 234 KB    | : 100% 1.0/1 [00:00<00:00, 10.72it/s]\n","pthread-stubs-0.4    | 5 KB      | : 100% 1.0/1 [00:00<00:00, 32.95it/s]\n","xorg-inputproto-2.3. | 19 KB     | : 100% 1.0/1 [00:00<00:00, 23.63it/s]\n","libzlib-1.2.11       | 59 KB     | : 100% 1.0/1 [00:00<00:00, 18.03it/s]\n","xorg-recordproto-1.1 | 8 KB      | : 100% 1.0/1 [00:00<00:00, 22.46it/s]\n","pcre2-10.35          | 693 KB    | : 100% 1.0/1 [00:00<00:00,  4.29it/s]\n","r-glue-1.4.2         | 141 KB    | : 100% 1.0/1 [00:00<00:00, 16.25it/s]\n","gcc_linux-64-7.5.0   | 23 KB     | : 100% 1.0/1 [00:00<00:00, 25.76it/s]\n","perl-time-local-1.28 | 10 KB     | : 100% 1.0/1 [00:00<00:00, 26.84it/s]\n","r-mgcv-1.8_33        | 2.9 MB    | : 100% 1.0/1 [00:00<00:00,  1.33it/s]\n","r-fansi-0.4.1        | 195 KB    | : 100% 1.0/1 [00:00<00:00,  3.48it/s]                \n","pyyaml-5.3.1         | 198 KB    | : 100% 1.0/1 [00:00<00:00, 13.80it/s]\n","llvm-openmp-11.0.0   | 2.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.96it/s]\n","pyrsistent-0.17.3    | 89 KB     | : 100% 1.0/1 [00:00<00:00, 18.15it/s]\n","pyzmq-20.0.0         | 485 KB    | : 100% 1.0/1 [00:00<00:00,  1.96it/s]\n","ipykernel-5.3.4      | 166 KB    | : 100% 1.0/1 [00:00<00:00,  9.49it/s]\n","q2-feature-classifie | 139 KB    | : 100% 1.0/1 [00:00<00:00,  2.61it/s]\n","joblib-0.17.0        | 204 KB    | : 100% 1.0/1 [00:00<00:00,  9.26it/s]\n","jinja2-2.11.2        | 93 KB     | : 100% 1.0/1 [00:00<00:00, 18.20it/s]\n","gfortran_impl_linux- | 9.0 MB    | : 100% 1.0/1 [00:01<00:00,  1.54s/it]               \n","tbb-2020.2           | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  3.66it/s]\n","pyopenssl-20.0.0     | 48 KB     | : 100% 1.0/1 [00:00<00:00, 15.19it/s]\n","perl-file-listing-6. | 9 KB      | : 100% 1.0/1 [00:00<00:00, 29.84it/s]\n","gfortran_linux-64-7. | 22 KB     | : 100% 1.0/1 [00:00<00:00, 11.77it/s]\n","perl-http-cookies-6. | 19 KB     | : 100% 1.0/1 [00:00<00:00, 25.48it/s]\n","packaging-20.7       | 32 KB     | : 100% 1.0/1 [00:00<00:00, 13.37it/s]\n","q2-dada2-2020.11.1   | 464 KB    | : 100% 1.0/1 [00:00<00:00,  2.40it/s]\n","attrs-20.3.0         | 41 KB     | : 100% 1.0/1 [00:00<00:00, 19.98it/s]\n","h5py-2.10.0          | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.57it/s]\n","pickleshare-0.7.5    | 9 KB      | : 100% 1.0/1 [00:00<00:00, 23.56it/s]\n","r-rstudioapi-0.13    | 281 KB    | : 100% 1.0/1 [00:00<00:00, 12.79it/s]\n","libcurl-7.71.1       | 312 KB    | : 100% 1.0/1 [00:00<00:00, 11.00it/s]\n","libffi-3.3           | 51 KB     | : 100% 1.0/1 [00:00<00:00, 22.39it/s]\n","async_generator-1.10 | 18 KB     | : 100% 1.0/1 [00:00<00:00, 25.39it/s]\n","r-permute-0.9_5      | 510 KB    | : 100% 1.0/1 [00:00<00:00,  9.78it/s]\n","glib-2.66.3          | 502 KB    | : 100% 1.0/1 [00:00<00:00,  2.04it/s]\n","r-farver-2.0.3       | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.81it/s]\n","readline-8.0         | 281 KB    | : 100% 1.0/1 [00:00<00:00, 10.66it/s]\n","hdf5-1.10.5          | 3.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.71it/s]\n","r-waldo-0.2.3        | 78 KB     | : 100% 1.0/1 [00:00<00:00,  4.08it/s]               \n","pluggy-0.13.1        | 29 KB     | : 100% 1.0/1 [00:00<00:00, 18.26it/s]\n","bioconductor-dada2-1 | 2.5 MB    | : 100% 1.0/1 [00:00<00:00,  2.74it/s]\n","r-matrixstats-0.57.0 | 918 KB    | : 100% 1.0/1 [00:00<00:00,  1.72it/s]\n","r-labeling-0.4.2     | 67 KB     | : 100% 1.0/1 [00:00<00:00, 21.56it/s]\n","c-ares-1.17.1        | 109 KB    | : 100% 1.0/1 [00:00<00:00, 16.02it/s]\n","r-hwriter-1.3.2      | 173 KB    | : 100% 1.0/1 [00:00<00:00, 19.51it/s]\n","perl-xml-namespacesu | 11 KB     | : 100% 1.0/1 [00:00<00:00, 22.05it/s]\n","samtools-1.11        | 383 KB    | : 100% 1.0/1 [00:00<00:00,  9.47it/s]\n","libglib-2.66.3       | 3.0 MB    | : 100% 1.0/1 [00:01<00:00,  1.10s/it]\n","numpy-1.19.4         | 5.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.78s/it]\n","r-backports-1.2.1    | 99 KB     | : 100% 1.0/1 [00:00<00:00, 21.02it/s]\n","perl-file-temp-0.230 | 26 KB     | : 100% 1.0/1 [00:00<00:00, 26.86it/s]\n","pigz-2.3.4           | 76 KB     | : 100% 1.0/1 [00:00<00:00, 20.58it/s]\n","perl-net-http-6.19   | 19 KB     | : 100% 1.0/1 [00:00<00:00, 24.47it/s]\n","biom-format-2.1.10   | 10.4 MB   | : 100% 1.0/1 [00:02<00:00,  2.73s/it]\n","_libgcc_mutex-0.1    | 3 KB      | : 100% 1.0/1 [00:00<00:00, 33.20it/s]\n","perl-5.26.2          | 15.4 MB   | : 100% 1.0/1 [00:04<00:00,  4.41s/it]              \n","perl-xsloader-0.24   | 8 KB      | : 100% 1.0/1 [00:00<00:00, 31.42it/s]\n","hdmedians-0.14.1     | 153 KB    | : 100% 1.0/1 [00:00<00:00,  3.09it/s]               \n","xorg-libxau-1.0.9    | 13 KB     | : 100% 1.0/1 [00:00<00:00, 29.59it/s]\n","bioconductor-rsamtoo | 3.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.70it/s]\n","perl-io-html-1.001   | 13 KB     | : 100% 1.0/1 [00:00<00:00, 23.32it/s]\n","q2-deblur-2020.11.1  | 20.4 MB   | : 100% 1.0/1 [00:03<00:00,  3.06s/it]\n","perl-storable-3.15   | 63 KB     | : 100% 1.0/1 [00:00<00:00, 22.46it/s]\n","perl-html-tree-5.07  | 129 KB    | : 100% 1.0/1 [00:00<00:00, 14.10it/s]\n","r-rcurl-1.98_1.2     | 960 KB    | : 100% 1.0/1 [00:00<00:00,  3.78it/s]               \n","perl-digest-hmac-1.0 | 9 KB      | : 100% 1.0/1 [00:00<00:00, 30.16it/s]\n","jupyterlab_pygments- | 8 KB      | : 100% 1.0/1 [00:00<00:00, 29.13it/s]\n","r-lifecycle-0.2.0    | 112 KB    | : 100% 1.0/1 [00:00<00:00,  3.00it/s]                \n","jupyter_core-4.7.0   | 72 KB     | : 100% 1.0/1 [00:00<00:00,  3.74it/s]                \n","lcms2-2.11           | 434 KB    | : 100% 1.0/1 [00:00<00:00,  6.74it/s]\n","perl-common-sense-3. | 11 KB     | : 100% 1.0/1 [00:00<00:00, 25.65it/s]\n","perl-io-zlib-1.10    | 9 KB      | : 100% 1.0/1 [00:00<00:00, 24.19it/s]\n","perl-parent-0.236    | 5 KB      | : 100% 1.0/1 [00:00<00:00, 29.03it/s]\n","q2-taxa-2020.11.1    | 108 KB    | : 100% 1.0/1 [00:00<00:00,  3.01it/s]\n","libgfortran-ng-7.5.0 | 23 KB     | : 100% 1.0/1 [00:00<00:00, 20.14it/s]\n","r-pkgbuild-1.1.0     | 156 KB    | : 100% 1.0/1 [00:00<00:00,  3.20it/s]                \n","toml-0.10.2          | 18 KB     | : 100% 1.0/1 [00:00<00:00, 20.49it/s]\n","r-jsonlite-1.7.2     | 462 KB    | : 100% 1.0/1 [00:00<00:00,  9.68it/s]\n","perl-xml-sax-1.02    | 36 KB     | : 100% 1.0/1 [00:00<00:00, 23.12it/s]\n","perl-io-socket-ssl-2 | 151 KB    | : 100% 1.0/1 [00:00<00:00, 13.96it/s]\n","xopen-1.0.1          | 19 KB     | : 100% 1.0/1 [00:00<00:00,  4.72it/s]               \n","krb5-1.17.2          | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.83it/s]\n","decorator-4.4.2      | 11 KB     | : 100% 1.0/1 [00:00<00:00, 29.26it/s]\n","perl-file-path-2.16  | 17 KB     | : 100% 1.0/1 [00:00<00:00, 29.82it/s]\n","r-testthat-3.0.0     | 1.6 MB    | : 100% 1.0/1 [00:00<00:00,  1.81it/s]\n","scikit-bio-0.5.6     | 1.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.91it/s]\n","dendropy-4.5.1       | 304 KB    | : 100% 1.0/1 [00:00<00:00,  6.69it/s]\n","zeromq-4.3.3         | 288 KB    | : 100% 1.0/1 [00:00<00:00, 11.97it/s]\n","perl-lwp-mediatypes- | 22 KB     | : 100% 1.0/1 [00:00<00:00, 24.87it/s]\n","bibtexparser-1.1.0   | 32 KB     | : 100% 1.0/1 [00:00<00:00, 24.06it/s]\n","patsy-0.5.1          | 187 KB    | : 100% 1.0/1 [00:00<00:00, 13.66it/s]\n","psutil-5.7.3         | 338 KB    | : 100% 1.0/1 [00:00<00:00,  2.13it/s]\n","q2-sample-classifier | 626 KB    | : 100% 1.0/1 [00:00<00:00,  2.06it/s]\n","bwidget-1.9.14       | 120 KB    | : 100% 1.0/1 [00:00<00:00, 12.10it/s]\n","giflib-5.2.1         | 77 KB     | : 100% 1.0/1 [00:00<00:00, 21.22it/s]\n","libopenblas-0.3.10   | 7.8 MB    | : 100% 1.0/1 [00:01<00:00,  1.65s/it]               \n","perl-www-robotrules- | 12 KB     | : 100% 1.0/1 [00:00<00:00, 21.08it/s]\n","blast-2.10.1         | 20.6 MB   | : 100% 1.0/1 [00:04<00:00,  4.27s/it]\n","pixman-0.40.0        | 627 KB    | : 100% 1.0/1 [00:00<00:00,  7.12it/s]\n","chardet-3.0.4        | 194 KB    | : 100% 1.0/1 [00:00<00:00,  6.02it/s]                \n","gsl-2.6              | 3.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.32it/s]\n","hmmer-3.1b2          | 5.5 MB    | : 100% 1.0/1 [00:00<00:00,  1.02it/s]\n","icu-67.1             | 12.9 MB   | : 100% 1.0/1 [00:02<00:00,  2.52s/it]               \n","r-ellipsis-0.3.1     | 49 KB     | : 100% 1.0/1 [00:00<00:00, 23.12it/s]\n","jpeg-9d              | 264 KB    | : 100% 1.0/1 [00:00<00:00, 10.74it/s]\n","xorg-libsm-1.2.3     | 26 KB     | : 100% 1.0/1 [00:00<00:00, 23.81it/s]\n","r-brio-1.1.0         | 37 KB     | : 100% 1.0/1 [00:00<00:00,  4.98it/s]               \n","r-futile.logger-1.4. | 108 KB    | : 100% 1.0/1 [00:00<00:00, 18.44it/s]\n","perl-exporter-5.72   | 13 KB     | : 100% 1.0/1 [00:00<00:00, 26.51it/s]\n","perl-http-message-6. | 51 KB     | : 100% 1.0/1 [00:00<00:00, 21.74it/s]\n","perl-http-negotiate- | 13 KB     | : 100% 1.0/1 [00:00<00:00, 29.20it/s]\n","q2-fragment-insertio | 31 KB     | : 100% 1.0/1 [00:00<00:00,  3.80it/s]\n","lz4-3.1.1            | 44 KB     | : 100% 1.0/1 [00:00<00:00,  3.75it/s]                \n","r-rlang-0.4.9        | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.03it/s]\n","nbclient-0.5.1       | 60 KB     | : 100% 1.0/1 [00:00<00:00,  4.38it/s]               \n","bioconductor-s4vecto | 2.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.34it/s]\n","r-munsell-0.5.0      | 247 KB    | : 100% 1.0/1 [00:00<00:00, 15.46it/s]\n","r-utf8-1.1.4         | 161 KB    | : 100% 1.0/1 [00:00<00:00,  3.30it/s]                \n","more-itertools-8.6.0 | 37 KB     | : 100% 1.0/1 [00:00<00:00, 16.45it/s]\n","bioconductor-biocpar | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.95it/s]\n","make-4.3             | 507 KB    | : 100% 1.0/1 [00:00<00:00,  6.34it/s]\n","cutadapt-3.1         | 184 KB    | : 100% 1.0/1 [00:00<00:00, 13.60it/s]\n","perl-list-moreutils- | 43 KB     | : 100% 1.0/1 [00:00<00:00, 23.16it/s]\n","r-rcppparallel-5.0.2 | 1.6 MB    | : 100% 1.0/1 [00:00<00:00,  2.40it/s]\n","perl-xml-sax-base-1. | 19 KB     | : 100% 1.0/1 [00:00<00:00, 24.48it/s]\n","py-1.9.0             | 74 KB     | : 100% 1.0/1 [00:00<00:00, 18.93it/s]\n","seaborn-0.11.0       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 18.51it/s]\n","tk-8.6.10            | 3.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.47it/s]\n","r-jpeg-0.1_8.1       | 52 KB     | : 100% 1.0/1 [00:00<00:00, 22.03it/s]\n","r-colorspace-2.0_0   | 2.5 MB    | : 100% 1.0/1 [00:00<00:00,  2.65it/s]\n","yaml-0.2.5           | 82 KB     | : 100% 1.0/1 [00:00<00:00, 21.45it/s]\n","graphite2-1.3.13     | 102 KB    | : 100% 1.0/1 [00:00<00:00, 20.40it/s]\n","setuptools-49.6.0    | 936 KB    | : 100% 1.0/1 [00:00<00:00,  3.53it/s]\n","backcall-0.2.0       | 13 KB     | : 100% 1.0/1 [00:00<00:00, 26.72it/s]\n","requests-2.25.0      | 51 KB     | : 100% 1.0/1 [00:00<00:00, 22.39it/s]\n","perl-io-compress-2.0 | 81 KB     | : 100% 1.0/1 [00:00<00:00, 13.68it/s]\n","sortmerna-2.0        | 337 KB    | : 100% 1.0/1 [00:00<00:00, 11.73it/s]\n","libgfortran4-7.5.0   | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  3.32it/s]\n","perl-types-serialise | 10 KB     | : 100% 1.0/1 [00:00<00:00, 22.91it/s]\n","ijson-3.1.3          | 24 KB     | : 100% 1.0/1 [00:00<00:00, 26.96it/s]\n","brotlipy-0.7.0       | 341 KB    | : 100% 1.0/1 [00:00<00:00,  9.96it/s]\n","jupyter_client-6.1.7 | 76 KB     | : 100% 1.0/1 [00:00<00:00, 12.06it/s]\n","cffi-1.14.4          | 225 KB    | : 100% 1.0/1 [00:00<00:00,  9.68it/s]\n","perl-xml-sax-expat-0 | 10 KB     | : 100% 1.0/1 [00:00<00:00, 28.30it/s]\n","pycparser-2.20       | 94 KB     | : 100% 1.0/1 [00:00<00:00, 14.28it/s]\n","libtiff-4.1.0        | 618 KB    | : 100% 1.0/1 [00:00<00:00,  6.49it/s]\n","argon2-cffi-20.1.0   | 47 KB     | : 100% 1.0/1 [00:00<00:00,  9.91it/s]\n","r-mass-7.3_53.1      | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  6.26it/s]\n","libiconv-1.16        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  4.05it/s]\n","libnghttp2-1.41.0    | 774 KB    | : 100% 1.0/1 [00:00<00:00,  5.35it/s]\n","xorg-libxi-1.7.10    | 45 KB     | : 100% 1.0/1 [00:00<00:00, 22.97it/s]\n","libuuid-2.32.1       | 28 KB     | : 100% 1.0/1 [00:00<00:00, 28.07it/s]\n","cython-0.29.21       | 2.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.03s/it]\n","gcc_impl_linux-64-7. | 38.3 MB   | : 100% 1.0/1 [00:07<00:00,  7.21s/it]               \n","ipython_genutils-0.2 | 21 KB     | : 100% 1.0/1 [00:00<00:00, 20.44it/s]\n","ca-certificates-2020 | 137 KB    | : 100% 1.0/1 [00:00<00:00, 19.13it/s]\n","libxml2-2.9.10       | 1.3 MB    | : 100% 1.0/1 [00:00<00:00,  2.21it/s]\n","seaborn-base-0.11.0  | 216 KB    | : 100% 1.0/1 [00:00<00:00,  7.64it/s]\n","r-snow-0.4_3         | 124 KB    | : 100% 1.0/1 [00:00<00:00, 20.15it/s]\n","r-latticeextra-0.6_2 | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  3.81it/s]\n","openjdk-11.0.8       | 173.3 MB  | : 100% 1.0/1 [00:24<00:00, 24.57s/it]\n","markupsafe-1.1.1     | 27 KB     | : 100% 1.0/1 [00:00<00:00,  5.14it/s]               \n","r-base-4.0.2         | 23.6 MB   | : 100% 1.0/1 [00:03<00:00,  3.68s/it]               \n","libgcc-7.2.0         | 304 KB    | : 100% 1.0/1 [00:00<00:00, 10.11it/s]\n","r-isoband-0.2.3      | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.65it/s]\n","liblapack-3.8.0      | 11 KB     | : 100% 1.0/1 [00:00<00:00, 29.41it/s]\n","perl-encode-2.88     | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.89it/s]\n","r-desc-1.2.0         | 291 KB    | : 100% 1.0/1 [00:00<00:00,  2.96it/s]                \n","r-pkgconfig-2.0.3    | 25 KB     | : 100% 1.0/1 [00:00<00:00, 26.97it/s]\n","q2-demux-2020.11.1   | 191 KB    | : 100% 1.0/1 [00:00<00:00,  2.94it/s]\n","click-7.1.2          | 64 KB     | : 100% 1.0/1 [00:00<00:00, 20.11it/s]\n","perl-xml-simple-2.25 | 48 KB     | : 100% 1.0/1 [00:00<00:00, 19.17it/s]\n","r-pkgload-1.1.0      | 167 KB    | : 100% 1.0/1 [00:00<00:00,  3.25it/s]                \n","ipywidgets-7.5.1     | 101 KB    | : 100% 1.0/1 [00:00<00:00, 16.16it/s]\n","harfbuzz-2.7.2       | 1.8 MB    | : 100% 1.0/1 [00:00<00:00,  2.49it/s]\n","pytz-2020.4          | 229 KB    | : 100% 1.0/1 [00:00<00:00,  5.46it/s]\n","perl-encode-locale-1 | 11 KB     | : 100% 1.0/1 [00:00<00:00, 22.84it/s]\n","r-rcpp-1.0.5         | 2.0 MB    | : 100% 1.0/1 [00:01<00:00,  1.00s/it]\n","nose-1.3.7           | 118 KB    | : 100% 1.0/1 [00:00<00:00, 15.44it/s]\n","pygments-2.7.3       | 708 KB    | : 100% 1.0/1 [00:00<00:00,  3.27it/s]\n","perl-lwp-protocol-ht | 9 KB      | : 100% 1.0/1 [00:00<00:00, 27.32it/s]\n","r-nlme-3.1_150       | 2.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.34it/s]\n","_openmp_mutex-4.5    | 5 KB      | : 100% 1.0/1 [00:00<00:00, 25.43it/s]\n","r-r6-2.5.0           | 89 KB     | : 100% 1.0/1 [00:00<00:00, 21.98it/s]\n","xorg-libice-1.0.10   | 58 KB     | : 100% 1.0/1 [00:00<00:00, 23.67it/s]\n","sysroot_linux-64-2.1 | 31.4 MB   | : 100% 1.0/1 [00:06<00:00,  6.10s/it]               \n","pillow-8.0.1         | 687 KB    | : 100% 1.0/1 [00:00<00:00,  5.06it/s]\n","gxx_linux-64-7.5.0   | 23 KB     | : 100% 1.0/1 [00:00<00:00, 25.67it/s]\n","perl-business-isbn-d | 15 KB     | : 100% 1.0/1 [00:00<00:00, 18.89it/s]\n","ncurses-6.2          | 985 KB    | : 100% 1.0/1 [00:00<00:00,  2.00it/s]\n","isa-l-2.30.0         | 192 KB    | : 100% 1.0/1 [00:00<00:00, 14.09it/s]\n","bioconductor-zlibbio | 113 KB    | : 100% 1.0/1 [00:00<00:00, 21.61it/s]\n","r-magrittr-2.0.1     | 211 KB    | : 100% 1.0/1 [00:00<00:00, 16.77it/s]\n","bowtie2-2.4.2        | 15.9 MB   | : 100% 1.0/1 [00:03<00:00,  3.52s/it]\n","ld_impl_linux-64-2.3 | 618 KB    | : 100% 1.0/1 [00:00<00:00,  5.21it/s]\n","q2cli-2020.11.1      | 77 KB     | : 100% 1.0/1 [00:00<00:00,  3.25it/s]\n","pango-1.42.4         | 533 KB    | : 100% 1.0/1 [00:00<00:00,  3.96it/s]\n","r-bh-1.72.0_3        | 10.5 MB   | : 100% 1.0/1 [00:06<00:00,  6.75s/it]\n","r-cli-2.2.0          | 443 KB    | : 100% 1.0/1 [00:00<00:00,  2.51it/s]\n","perl-try-tiny-0.30   | 17 KB     | : 100% 1.0/1 [00:00<00:00, 25.46it/s]\n","perl-extutils-makema | 153 KB    | : 100% 1.0/1 [00:00<00:00, 11.44it/s]\n","threadpoolctl-2.1.0  | 15 KB     | : 100% 1.0/1 [00:00<00:00, 26.92it/s]\n","urllib3-1.25.11      | 93 KB     | : 100% 1.0/1 [00:00<00:00, 16.05it/s]\n","perl-carp-1.38       | 15 KB     | : 100% 1.0/1 [00:00<00:00, 27.95it/s]\n","jedi-0.17.2          | 957 KB    | : 100% 1.0/1 [00:00<00:00,  1.61it/s]\n","lz4-c-1.9.2          | 203 KB    | : 100% 1.0/1 [00:00<00:00, 12.80it/s]\n","prompt-toolkit-3.0.8 | 237 KB    | : 100% 1.0/1 [00:00<00:00,  5.71it/s]\n","q2-quality-filter-20 | 32 KB     | : 100% 1.0/1 [00:00<00:00,  3.95it/s]\n","ptyprocess-0.6.0     | 15 KB     | : 100% 1.0/1 [00:00<00:00, 16.27it/s]\n","gneiss-0.4.6         | 41 KB     | : 100% 1.0/1 [00:00<00:00, 23.19it/s]\n","sina-1.7.1           | 1.9 MB    | : 100% 1.0/1 [00:00<00:00,  1.13it/s]\n","r-futile.options-1.0 | 27 KB     | : 100% 1.0/1 [00:00<00:00, 21.19it/s]\n","tornado-6.1          | 643 KB    | : 100% 1.0/1 [00:00<00:00,  4.34it/s]\n","prometheus_client-0. | 44 KB     | : 100% 1.0/1 [00:00<00:00, 13.71it/s]\n","libev-4.33           | 104 KB    | : 100% 1.0/1 [00:00<00:00, 17.79it/s]\n","q2-diversity-lib-202 | 88 KB     | : 100% 1.0/1 [00:00<00:00,  3.14it/s]\n","certifi-2020.12.5    | 143 KB    | : 100% 1.0/1 [00:00<00:00,  3.29it/s]                \n","perl-archive-tar-2.3 | 31 KB     | : 100% 1.0/1 [00:00<00:00, 24.00it/s]\n","bioconductor-genomei | 7 KB      | : 100% 1.0/1 [00:00<00:00,  5.82it/s]\n","expat-2.2.9          | 163 KB    | : 100% 1.0/1 [00:00<00:00, 10.87it/s]\n","libarbdb-6.0.6       | 290 KB    | : 100% 1.0/1 [00:00<00:00,  2.73it/s]                \n","r-crayon-1.3.4       | 747 KB    | : 100% 1.0/1 [00:00<00:00,  5.16it/s]\n","perl-mozilla-ca-2018 | 140 KB    | : 100% 1.0/1 [00:00<00:00, 19.00it/s]\n","r-withr-2.3.0        | 216 KB    | : 100% 1.0/1 [00:00<00:00,  3.20it/s]                \n","perl-scalar-list-uti | 38 KB     | : 100% 1.0/1 [00:00<00:00, 21.58it/s]\n","widgetsnbextension-3 | 1.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.99it/s]\n","bleach-3.2.1         | 111 KB    | : 100% 1.0/1 [00:00<00:00, 15.05it/s]\n","webencodings-0.5.1   | 12 KB     | : 100% 1.0/1 [00:00<00:00, 30.55it/s]\n","q2-gneiss-2020.11.1  | 50 KB     | : 100% 1.0/1 [00:00<00:00,  3.61it/s]\n","fontconfig-2.13.1    | 357 KB    | : 100% 1.0/1 [00:00<00:00,  8.77it/s]\n","backports-1.0        | 4 KB      | : 100% 1.0/1 [00:00<00:00, 31.61it/s]\n","perl-business-isbn-3 | 15 KB     | : 100% 1.0/1 [00:00<00:00, 18.88it/s]\n","dnaio-0.4.4          | 136 KB    | : 100% 1.0/1 [00:00<00:00, 17.16it/s]\n","r-digest-0.6.27      | 211 KB    | : 100% 1.0/1 [00:00<00:00, 13.24it/s]\n","r-viridislite-0.3.0  | 64 KB     | : 100% 1.0/1 [00:00<00:00,  4.48it/s]                \n","perl-test-requiresin | 7 KB      | : 100% 1.0/1 [00:00<00:00, 30.32it/s]\n","python_abi-3.6       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 28.86it/s]\n","sqlite-3.34.0        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  3.45it/s]\n","binutils_impl_linux- | 9.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.78s/it]               \n","bioconductor-biobase | 2.3 MB    | : 100% 1.0/1 [00:00<00:00,  2.81it/s]\n","perl-uri-1.76        | 55 KB     | : 100% 1.0/1 [00:00<00:00, 20.73it/s]\n","bioconductor-matrixg | 321 KB    | : 100% 1.0/1 [00:00<00:00, 10.15it/s]\n","networkx-2.5.1       | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  2.50it/s]\n","r-vctrs-0.3.5        | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.10it/s]\n","msgpack-python-1.0.0 | 91 KB     | : 100% 1.0/1 [00:00<00:00, 13.46it/s]\n","pyparsing-2.4.7      | 60 KB     | : 100% 1.0/1 [00:00<00:00, 20.06it/s]\n","xorg-fixesproto-5.0  | 9 KB      | : 100% 1.0/1 [00:00<00:00, 30.03it/s]\n","iniconfig-1.1.1      | 8 KB      | : 100% 1.0/1 [00:00<00:00, 29.14it/s]\n","natsort-7.1.0        | 33 KB     | : 100% 1.0/1 [00:00<00:00,  3.44it/s]               \n","r-plyr-1.8.6         | 832 KB    | : 100% 1.0/1 [00:00<00:00,  6.98it/s]\n","vsearch-2.7.0        | 713 KB    | : 100% 1.0/1 [00:00<00:00,  5.95it/s]\n","perl-exporter-tiny-1 | 23 KB     | : 100% 1.0/1 [00:00<00:00, 26.81it/s]\n","r-bitops-1.0_6       | 39 KB     | : 100% 1.0/1 [00:00<00:00, 22.55it/s]\n","libcblas-3.8.0       | 11 KB     | : 100% 1.0/1 [00:00<00:00, 25.22it/s]\n","r-scales-1.1.1       | 556 KB    | : 100% 1.0/1 [00:00<00:00,  9.73it/s]\n","traitlets-4.3.3      | 62 KB     | : 100% 1.0/1 [00:00<00:00, 18.67it/s]\n","libstdcxx-ng-9.3.0   | 4.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.30it/s]\n","r-gtable-0.3.0       | 416 KB    | : 100% 1.0/1 [00:00<00:00,  8.29it/s]\n","libedit-3.1.20191231 | 121 KB    | : 100% 1.0/1 [00:00<00:00, 18.65it/s]\n","xorg-libxext-1.3.4   | 51 KB     | : 100% 1.0/1 [00:00<00:00, 19.78it/s]\n","wheel-0.36.1         | 30 KB     | : 100% 1.0/1 [00:00<00:00, 17.24it/s]\n","libblas-3.8.0        | 11 KB     | : 100% 1.0/1 [00:00<00:00, 27.88it/s]\n","freetype-2.10.4      | 890 KB    | : 100% 1.0/1 [00:00<00:00,  4.07it/s]\n","q2templates-2020.11. | 188 KB    | : 100% 1.0/1 [00:00<00:00,  2.93it/s]\n","r-pillar-1.4.7       | 194 KB    | : 100% 1.0/1 [00:00<00:00,  3.03it/s]                \n","perl-html-parser-3.7 | 66 KB     | : 100% 1.0/1 [00:00<00:00, 17.78it/s]\n","q2-quality-control-2 | 285 KB    | : 100% 1.0/1 [00:00<00:00,  2.76it/s]\n","xorg-kbproto-1.0.7   | 27 KB     | : 100% 1.0/1 [00:00<00:00, 24.45it/s]\n","xorg-renderproto-0.1 | 9 KB      | : 100% 1.0/1 [00:00<00:00, 28.21it/s]\n","mafft-7.475          | 3.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.48it/s]\n","perl-json-4.02       | 52 KB     | : 100% 1.0/1 [00:00<00:00, 20.80it/s]\n","mistune-0.8.4        | 54 KB     | : 100% 1.0/1 [00:00<00:00, 16.40it/s]\n","perl-http-date-6.02  | 10 KB     | : 100% 1.0/1 [00:00<00:00, 28.05it/s]\n","q2-vsearch-2020.11.1 | 634 KB    | : 100% 1.0/1 [00:00<00:00,  2.09it/s]\n","pandoc-2.11.2        | 17.7 MB   | : 100% 1.0/1 [00:05<00:00,  5.58s/it]               \n","perl-base-2.23       | 10 KB     | : 100% 1.0/1 [00:00<00:00, 31.25it/s]\n","raxml-8.2.12         | 3.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.89it/s]\n","pcre-8.44            | 261 KB    | : 100% 1.0/1 [00:00<00:00, 11.06it/s]\n","boost-cpp-1.70.0     | 21.1 MB   | : 100% 1.0/1 [00:09<00:00,  9.03s/it]               \n","tzlocal-2.1          | 18 KB     | : 100% 1.0/1 [00:00<00:00, 27.41it/s]\n","fastcluster-1.1.26   | 45 KB     | : 100% 1.0/1 [00:00<00:00,  4.17it/s]                \n","backports.functools_ | 8 KB      | : 100% 1.0/1 [00:00<00:00, 30.01it/s]\n","nbconvert-6.0.7      | 533 KB    | : 100% 1.0/1 [00:00<00:00,  5.16it/s]\n","deblur-1.1.0         | 2.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.02s/it]\n","libxcb-1.13          | 395 KB    | : 100% 1.0/1 [00:00<00:00,  5.83it/s]\n","importlib-metadata-3 | 12 KB     | : 100% 1.0/1 [00:00<00:00, 16.13it/s]\n","perl-socket-2.027    | 31 KB     | : 100% 1.0/1 [00:00<00:00, 21.68it/s]\n","bioconductor-iranges | 2.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.33it/s]\n","entrypoints-0.3      | 8 KB      | : 100% 1.0/1 [00:00<00:00, 29.64it/s]\n","nbformat-5.0.8       | 100 KB    | : 100% 1.0/1 [00:00<00:00,  1.39it/s]\n","r-lattice-0.20_41    | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  5.63it/s]\n","perl-compress-raw-zl | 72 KB     | : 100% 1.0/1 [00:00<00:00, 20.42it/s]\n","kiwisolver-1.3.1     | 78 KB     | : 100% 1.0/1 [00:00<00:00, 21.14it/s]\n","bioconductor-genomic | 2.2 MB    | : 100% 1.0/1 [00:00<00:00,  2.84it/s]\n","bioconductor-biostri | 13.9 MB   | : 100% 1.0/1 [00:01<00:00,  1.58s/it]               \n","jsonschema-3.2.0     | 45 KB     | : 100% 1.0/1 [00:00<00:00, 17.40it/s]\n","r-reshape2-1.4.4     | 128 KB    | : 100% 1.0/1 [00:00<00:00, 17.70it/s]\n","zstd-1.4.5           | 712 KB    | : 100% 1.0/1 [00:00<00:00,  6.41it/s]\n","perl-list-moreutils- | 28 KB     | : 100% 1.0/1 [00:00<00:00, 24.15it/s]\n","cairo-1.16.0         | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  2.64it/s]\n","libpng-1.6.37        | 306 KB    | : 100% 1.0/1 [00:00<00:00,  9.47it/s]\n","wcwidth-0.2.5        | 33 KB     | : 100% 1.0/1 [00:00<00:00, 21.49it/s]\n","zlib-1.2.11          | 86 KB     | : 100% 1.0/1 [00:00<00:00, 20.30it/s]\n","zipp-3.4.0           | 11 KB     | : 100% 1.0/1 [00:00<00:00, 18.51it/s]\n","q2-feature-table-202 | 227 KB    | : 100% 1.0/1 [00:00<00:00,  2.46it/s]\n","r-vegan-2.5_6        | 3.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.03it/s]\n","r-cluster-2.1.0      | 553 KB    | : 100% 1.0/1 [00:00<00:00,  2.44it/s]\n","scikit-learn-0.23.1  | 6.8 MB    | : 100% 1.0/1 [00:02<00:00,  2.06s/it]               \n","qiime2-2020.11.1     | 320 KB    | : 100% 1.0/1 [00:00<00:00,  1.91it/s]\n","r-evaluate-0.14      | 81 KB     | : 100% 1.0/1 [00:00<00:00, 19.40it/s]\n","r-callr-3.5.1        | 388 KB    | : 100% 1.0/1 [00:00<00:00,  2.74it/s]                 \n","statsmodels-0.12.1   | 11.4 MB   | : 100% 1.0/1 [00:03<00:00,  3.47s/it]\n","blas-2.17            | 11 KB     | : 100% 1.0/1 [00:00<00:00, 27.21it/s]\n","r-lambda.r-1.2.4     | 120 KB    | : 100% 1.0/1 [00:00<00:00, 21.12it/s]\n","xorg-libx11-1.6.12   | 919 KB    | : 100% 1.0/1 [00:00<00:00,  3.84it/s]\n","pip-20.3.1           | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.87it/s]\n","perl-ntlm-1.09       | 15 KB     | : 100% 1.0/1 [00:00<00:00, 28.06it/s]\n","htslib-1.11          | 1.7 MB    | : 100% 1.0/1 [00:00<00:00,  2.66it/s]\n","parso-0.7.1          | 70 KB     | : 100% 1.0/1 [00:00<00:00, 17.68it/s]\n","cachecontrol-0.12.6  | 18 KB     | : 100% 1.0/1 [00:00<00:00, 23.97it/s]\n","libgcc-ng-9.3.0      | 7.8 MB    | : 100% 1.0/1 [00:01<00:00,  1.46s/it]               \n","perl-net-ssleay-1.88 | 289 KB    | : 100% 1.0/1 [00:00<00:00,  8.80it/s]\n","testpath-0.4.4       | 85 KB     | : 100% 1.0/1 [00:00<00:00, 22.98it/s]\n","q2-diversity-2020.11 | 225 KB    | : 100% 1.0/1 [00:00<00:00,  2.75it/s]\n","matplotlib-3.2.2     | 6 KB      | : 100% 1.0/1 [00:00<00:00, 30.14it/s]\n","r-prettyunits-1.1.1  | 41 KB     | : 100% 1.0/1 [00:00<00:00, 25.42it/s]\n","libsodium-1.0.18     | 366 KB    | : 100% 1.0/1 [00:00<00:00,  8.85it/s]\n","q2-metadata-2020.11. | 967 KB    | : 100% 1.0/1 [00:00<00:00,  2.02it/s]\n","openssl-1.1.1h       | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.40it/s]\n","iqtree-2.0.3         | 3.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.23it/s]\n","perl-html-tagset-3.2 | 12 KB     | : 100% 1.0/1 [00:00<00:00, 23.72it/s]\n","matplotlib-base-3.2. | 7.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.70s/it]\n","fasttree-2.1.10      | 477 KB    | : 100% 1.0/1 [00:00<00:00,  7.04it/s]\n","future-0.18.2        | 716 KB    | : 100% 1.0/1 [00:00<00:00,  3.32it/s]\n","r-stringi-1.5.3      | 805 KB    | : 100% 1.0/1 [00:00<00:00,  2.35it/s]\n","importlib_metadata-3 | 3 KB      | : 100% 1.0/1 [00:00<00:00, 19.45it/s]\n","r-formatr-1.7        | 166 KB    | : 100% 1.0/1 [00:00<00:00,  3.17it/s]                \n","emperor-1.0.1        | 621 KB    | : 100% 1.0/1 [00:00<00:00,  4.74it/s]\n","perl-compress-raw-bz | 44 KB     | : 100% 1.0/1 [00:00<00:00, 23.32it/s]\n","unifrac-0.20.1       | 356 KB    | : 100% 1.0/1 [00:00<00:00,  2.63it/s]\n","xorg-libxdmcp-1.1.3  | 19 KB     | : 100% 1.0/1 [00:00<00:00, 25.41it/s]\n","xorg-xproto-7.0.31   | 73 KB     | : 100% 1.0/1 [00:00<00:00, 20.96it/s]\n","xorg-xextproto-7.3.0 | 28 KB     | : 100% 1.0/1 [00:00<00:00, 24.38it/s]\n","defusedxml-0.6.0     | 22 KB     | : 100% 1.0/1 [00:00<00:00, 18.58it/s]\n","r-tibble-3.0.4       | 384 KB    | : 100% 1.0/1 [00:00<00:00,  8.48it/s]\n","libstdcxx-devel_linu | 9.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.89s/it]               \n","perl-json-xs-2.34    | 63 KB     | : 100% 1.0/1 [00:00<00:00, 20.04it/s]\n","r-matrix-1.2_18      | 3.8 MB    | : 100% 1.0/1 [00:01<00:00,  1.17s/it]\n","lockfile-0.12.2      | 11 KB     | : 100% 1.0/1 [00:00<00:00, 30.00it/s]\n","arb-bio-tools-6.0.6  | 673 KB    | : 100% 1.0/1 [00:00<00:00,  2.04it/s]\n","Preparing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","Executing transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- Enabling notebook extension jupyter-js-widgets/extension...\n","Paths used for configuration of notebook: \n","    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.d/widgetsnbextension.json\n","    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.json\n","Paths used for configuration of notebook: \n","    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.d/widgetsnbextension.json\n","      - Validating: \u001b[32mOK\u001b[0m\n","Paths used for configuration of notebook: \n","    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.d/widgetsnbextension.json\n","    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.json\n","\n","\b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","#\n","# To activate this environment, use\n","#\n","#     $ conda activate qiime2-2020.11\n","#\n","# To deactivate an active environment, use\n","#\n","#     $ conda deactivate\n","\n"]}]},{"cell_type":"markdown","metadata":{"id":"39M8EoI-Ahie"},"source":["#Test Qiime2, all qiime2 commands need to be executed with \"!source activate qiime2-2020.11 &&\""]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"nQW9Fd0E4v23","executionInfo":{"status":"ok","timestamp":1630551260031,"user_tz":-60,"elapsed":3264,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"3f05cea0-e7f6-4d38-897a-703760bc24ac"},"source":["!source activate qiime2-2020.11 && qiime --help"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime\u001b[0m [OPTIONS] COMMAND [ARGS]...\n","\n","  QIIME 2 command-line interface (q2cli)\n","  --------------------------------------\n","\n","  To get help with QIIME 2, visit\n","  https://qiime2.org.\n","\n","  To enable tab completion in Bash, run the\n","  following command or add it to your\n","  .bashrc/.bash_profile:\n","\n","      source tab-qiime\n","\n","  To enable tab completion in ZSH, run the\n","  following commands or add them to your .zshrc:\n","\n","      autoload bashcompinit && bashcompinit &&\n","      source tab-qiime\n","\n","\u001b[1mOptions\u001b[0m:\n","  \u001b[34m--version\u001b[0m   Show the version and exit.\n","  \u001b[34m--help\u001b[0m      Show this message and exit.\n","\n","\u001b[1mCommands\u001b[0m:\n","  \u001b[34minfo\u001b[0m                Display information about...\n","  \u001b[34mtools\u001b[0m               Tools for working with...\n","  \u001b[34mdev\u001b[0m                 Utilities for developers...\n","  \u001b[34malignment\u001b[0m           Plugin for generating and\n","                      manipulating alignments.\n","\n","  \u001b[34mcomposition\u001b[0m         Plugin for compositional\n","                      data analysis.\n","\n","  \u001b[34mcutadapt\u001b[0m            Plugin for removing adapter\n","                      sequences, primers, and\n","                      other unwanted sequence from\n","                      sequence data.\n","\n","  \u001b[34mdada2\u001b[0m               Plugin for sequence quality\n","                      control with DADA2.\n","\n","  \u001b[34mdeblur\u001b[0m              Plugin for sequence quality\n","                      control with Deblur.\n","\n","  \u001b[34mdemux\u001b[0m               Plugin for demultiplexing &\n","                      viewing sequence quality.\n","\n","  \u001b[34mdiversity\u001b[0m           Plugin for exploring\n","                      community diversity.\n","\n","  \u001b[34mdiversity-lib\u001b[0m       Plugin for computing\n","                      community diversity.\n","\n","  \u001b[34memperor\u001b[0m             Plugin for ordination\n","                      plotting with Emperor.\n","\n","  \u001b[34mfeature-classifier\u001b[0m  Plugin for taxonomic\n","                      classification.\n","\n","  \u001b[34mfeature-table\u001b[0m       Plugin for working with\n","                      sample by feature tables.\n","\n","  \u001b[34mfragment-insertion\u001b[0m  Plugin for extending\n","                      phylogenies.\n","\n","  \u001b[34mgneiss\u001b[0m              Plugin for building\n","                      compositional models.\n","\n","  \u001b[34mlongitudinal\u001b[0m        Plugin for paired sample and\n","                      time series analyses.\n","\n","  \u001b[34mmetadata\u001b[0m            Plugin for working with\n","                      Metadata.\n","\n","  \u001b[34mphylogeny\u001b[0m           Plugin for generating and\n","                      manipulating phylogenies.\n","\n","  \u001b[34mquality-control\u001b[0m     Plugin for quality control\n","                      of feature and sequence\n","                      data.\n","\n","  \u001b[34mquality-filter\u001b[0m      Plugin for PHRED-based\n","                      filtering and trimming.\n","\n","  \u001b[34msample-classifier\u001b[0m   Plugin for machine learning\n","                      prediction of sample\n","                      metadata.\n","\n","  \u001b[34mtaxa\u001b[0m                Plugin for working with\n","                      feature taxonomy\n","                      annotations.\n","\n","  \u001b[34mvsearch\u001b[0m             Plugin for clustering and\n","                      dereplicating with vsearch.\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"tKMJJP-l9wvm","executionInfo":{"status":"ok","timestamp":1630551260032,"user_tz":-60,"elapsed":10,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"d8686e44-a8c9-4725-8f14-12c47ee88050"},"source":["cd /content/sample_data/"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/sample_data\n"]}]},{"cell_type":"markdown","metadata":{"id":"XTTzjmsg6WYr"},"source":["#**STOP HERE** goto step 2"]},{"cell_type":"markdown","metadata":{"id":"cF-keGhzH4oK"},"source":["#Create a manifest for the data"]},{"cell_type":"code","metadata":{"id":"nPhfyE3lUfI7"},"source":["!cp /content/sample_data/*.qz* /content/drive/MyDrive\\ "],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"U3yCpFk0CUlX"},"source":["#Import data to Qiime2"]},{"cell_type":"markdown","metadata":{"id":"yZdaIMj3vN8D"},"source":["#Remove underscores from file name"]},{"cell_type":"markdown","metadata":{"id":"dwaI7mukvT-M"},"source":["#Create a manifest"]},{"cell_type":"code","metadata":{"id":"CNZyH96d-X8V"},"source":["!echo -e \"sample-id\\tforward-absolute-filepath\\treverse-absolute-filepath\" > manifest.tsv"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"4sb4RORr-fcG","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1630552277141,"user_tz":-60,"elapsed":271,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"b96964b4-1ff5-4732-f4d1-18d16ae45688"},"source":["%%bash\n","for f in `ls *fastq.gz | cut -d '_' -f 1 | sort -u`; do n=`basename $f`; echo -e \"${n}\\t$PWD/$f\"_R1_001.fastq.gz\"\\t$PWD/$f\"_R2_001.fastq.gz\" \"; done >> manifest.tsv "],"execution_count":null,"outputs":[{"output_type":"stream","name":"stderr","text":["ls: cannot access '*fastq.gz': No such file or directory\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"VkRWOEVG-oFP","executionInfo":{"status":"ok","timestamp":1630552287437,"user_tz":-60,"elapsed":7744,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"1f7802c7-1dff-4042-e426-3f7ed87797fe"},"source":["!source activate qiime2-2020.11 && qiime tools import \\\n","  --input-path manifest.tsv \\\n","  --type 'SampleData[PairedEndSequencesWithQuality]' \\\n","  --input-format PairedEndFastqManifestPhred33V2 \\\n","  --output-path pe-demux.qza"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[31m\u001b[1mThere was a problem importing manifest.tsv:\n","\n","  manifest.tsv is not a(n) PairedEndFastqManifestPhred33V2 file:\n","\n","  There was an issue with loading the metadata file:\n","\n","  Metadata must contain at least one ID.\n","\n","  There may be more errors present in the metadata file. To get a full report, sample/feature metadata files can be validated with Keemei: https://keemei.qiime2.org\n","\n","  Find details on QIIME 2 metadata requirements here: https://docs.qiime2.org/2020.11/tutorials/metadata/\u001b[0m\n","\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"L0ok57E39ts5","outputId":"e4aa1925-57e1-4b7e-95e8-4171141bb4f2"},"source":["!source activate qiime2-2020.11 && qiime vsearch join-pairs \\\n","  --i-demultiplexed-seqs pe-demux.qza \\\n","  --o-joined-sequences demux-joined.qza"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\u001b[32mSaved SampleData[JoinedSequencesWithQuality] to: demux-joined.qza\u001b[0m\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"uqhaHa9Y-oUn","outputId":"ab505758-dbe0-44c6-c4f2-de3f9ba39eb0"},"source":["!source activate qiime2-2020.11 && qiime demux summarize \\\n","    --i-data  demux-joined.qza \\\n","    --o-visualization demux-joined.qzv"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\u001b[32mSaved Visualization to: demux-joined.qzv\u001b[0m\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"5WyjQ-vh-oe6","outputId":"dcf7b7d8-bc89-4d66-992f-071b01871c61"},"source":["!source activate qiime2-2020.11 && qiime quality-filter q-score \\\n","    --i-demux demux-joined.qza \\\n","    --o-filtered-sequences demux-filtered.qza \\\n","    --o-filter-stats demux-filter-stats.qza"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\u001b[32mSaved SampleData[JoinedSequencesWithQuality] to: demux-filtered.qza\u001b[0m\n","\u001b[32mSaved QualityFilterStats to: demux-filter-stats.qza\u001b[0m\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"rBS9FJcS803Q"},"source":["!cp /content/sample_data/*.qz* /content/drive/MyDrive/AlbertFrench\\ "],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"IDoMH4cH3i49"},"source":["#DOWNLOAD demux-joined.qzv UPLOAD ON QIIME VIEW https://view.qiime2.org/"]},{"cell_type":"code","metadata":{"id":"2yvEMCp8WrIb"},"source":["cd /content/sample_data"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"8HWr4KtgUDIr"},"source":["#Denoise steps using deblur, trim length is from the demux-joined.qzv file\n"]},{"cell_type":"code","metadata":{"id":"jrsSHzok-oox"},"source":["!source activate qiime2-2020.11 && qiime deblur denoise-16S \\\n","    --i-demultiplexed-seqs demux-filtered.qza \\\n","    --p-trim-length 200 \\\n","    --p-sample-stats \\\n","    --o-stats deblur-stats.qza \\\n","    --o-representative-sequences rep-seqs-deblur.qza \\\n","    --o-table table-deblur.qza"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"JOlmSsY6-ozc","outputId":"d73f2467-a646-47ed-b15f-cc1787864cc6"},"source":["!source activate qiime2-2020.11 && qiime deblur visualize-stats \\\n","    --i-deblur-stats deblur-stats.qza \\\n","    --o-visualization deblur-stats.qzv"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\u001b[32mSaved Visualization to: deblur-stats.qzv\u001b[0m\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"vUVbVXuW-o-B","outputId":"4a017295-f213-4e22-810e-3381bf7270de"},"source":["!source activate qiime2-2020.11 && qiime feature-table tabulate-seqs \\\n","    --i-data rep-seqs-deblur.qza \\\n","    --o-visualization rep-seqs-deblur.qzv"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\u001b[32mSaved Visualization to: rep-seqs-deblur.qzv\u001b[0m\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"ifXflYXssRSF"},"source":["#**2.start here again**"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"eQG1M1WT-_VR","executionInfo":{"status":"ok","timestamp":1630551612850,"user_tz":-60,"elapsed":4225,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"b89612fe-d38b-417a-f385-35eff86c4520"},"source":["!source activate qiime2-2020.11 && qiime feature-table summarize \\\n","    --i-table table-deblur.qza \\\n","    --m-sample-metadata-file AFmetadata.txt \\\n","    --o-visualization table-deblur.qzv"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime feature-table summarize\u001b[0m \n","           [OPTIONS]\n","\n","  Generate visual and tabular summaries of a\n","  feature table.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-table\u001b[0m ARTIFACT \u001b[32mFeatureTable[Frequency |\u001b[0m\n","    \u001b[32mRelativeFrequency | PresenceAbsence]\u001b[0m\n","                       The feature table to be\n","                       summarized.      \u001b[35m[required]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m--m-sample-metadata-file\u001b[0m METADATA...\n","    (multiple          The sample metadata.\n","     arguments will    \n","     be merged)                         \u001b[35m[optional]\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-visualization\u001b[0m VISUALIZATION\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH    Output unspecified results\n","                       to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m  Display verbose output to\n","                       stdout and/or stderr during\n","                       execution of this action.\n","                       Or silence output if\n","                       execution is successful\n","                       (silence is golden).\n","  \u001b[34m--examples\u001b[0m           Show usage examples and\n","                       exit.\n","  \u001b[34m--citations\u001b[0m          Show citations and exit.\n","  \u001b[34m--help\u001b[0m               Show this message and\n","                       exit.\n","\n","\u001b[33m                    There was a problem with the command:                     \u001b[0m\n","\u001b[31m\u001b[1m (1/1) Invalid value for '--i-table': 'table-deblur.qza' is not a valid\n","  filepath\u001b[0m\n"]}]},{"cell_type":"markdown","metadata":{"id":"wIexsQt8_y2G"},"source":["#Phylogenic tree (takes a few min)"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"-JgHT4KQvCoF","executionInfo":{"status":"ok","timestamp":1630551267819,"user_tz":-60,"elapsed":3871,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"6f13f0ef-7c21-4193-e4c7-0370a32cb0f9"},"source":["!source activate qiime2-2020.11 && qiime phylogeny align-to-tree-mafft-fasttree \\\n","  --i-sequences rep-seqs-deblur.qza  \\\n","  --o-alignment aligned-rep-seqs.qza \\\n","  --o-masked-alignment masked-aligned-rep-seqs.qza \\\n","  --o-tree unrooted-tree.qza \\\n","  --o-rooted-tree rooted-tree.qza"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime phylogeny align-to-tree-mafft-fasttree\u001b[0m \n","           [OPTIONS]\n","\n","  This pipeline will start by creating a\n","  sequence alignment using MAFFT, after which\n","  any alignment columns that are\n","  phylogenetically uninformative or ambiguously\n","  aligned will be removed (masked). The\n","  resulting masked alignment will be used to\n","  infer a phylogenetic tree and then\n","  subsequently rooted at its midpoint. Output\n","  files from each step of the pipeline will be\n","  saved. This includes both the unmasked and\n","  masked MAFFT alignment from q2-alignment\n","  methods, and both the rooted and unrooted\n","  phylogenies from q2-phylogeny methods.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-sequences\u001b[0m ARTIFACT \u001b[32mFeatureData[Sequence]\u001b[0m\n","                          The sequences to be\n","                          used for creating a\n","                          fasttree based rooted\n","                          phylogenetic tree.\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m--p-n-threads\u001b[0m VALUE \u001b[32mInt % Range(1, None) |\u001b[0m\n","    \u001b[32mStr % Choices('auto')\u001b[0m The number of threads.\n","                          (Use `auto` to\n","                          automatically use all\n","                          available cores) This\n","                          value is used when\n","                          aligning the sequences\n","                          and creating the tree\n","                          with fasttree.\n","                                      \u001b[35m[default: 1]\u001b[0m\n","  \u001b[34m--p-mask-max-gap-frequency\u001b[0m PROPORTION \u001b[32m\u001b[0m\n","    \u001b[32mRange(0, 1, inclusive_end=True)\u001b[0m\n","                          The maximum relative\n","                          frequency of gap\n","                          characters in a column\n","                          for the column to be\n","                          retained. This relative\n","                          frequency must be a\n","                          number between 0.0 and\n","                          1.0 (inclusive), where\n","                          0.0 retains only those\n","                          columns without gap\n","                          characters, and 1.0\n","                          retains all columns \n","                          regardless of gap\n","                          character frequency.\n","                          This value is used when\n","                          masking the aligned\n","                          sequences.\n","                                    \u001b[35m[default: 1.0]\u001b[0m\n","  \u001b[34m--p-mask-min-conservation\u001b[0m PROPORTION \u001b[32mRange(0,\u001b[0m\n","    \u001b[32m1, inclusive_end=True)\u001b[0m\n","                          The minimum relative\n","                          frequency of at least\n","                          one non-gap character in\n","                          a column for that column\n","                          to be retained. This\n","                          relative frequency must\n","                          be a number between 0.0\n","                          and 1.0 (inclusive). For\n","                          example, if a value of \n","                          0.4 is provided, a\n","                          column will only be\n","                          retained  if it contains\n","                          at least one character\n","                          that is present in at\n","                          least 40% of the\n","                          sequences. This value is\n","                          used when masking the\n","                          aligned sequences.\n","                                    \u001b[35m[default: 0.4]\u001b[0m\n","  \u001b[34m--p-parttree\u001b[0m / \u001b[34m--p-no-parttree\u001b[0m\n","                          This flag is required\n","                          if the number of\n","                          sequences being aligned\n","                          are larger than 1000000.\n","                          Disabled by default.\n","                                  \u001b[35m[default: False]\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-alignment\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mFeatureData[AlignedSequence]\u001b[0m\n","                          The aligned sequences.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-masked-alignment\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mFeatureData[AlignedSequence]\u001b[0m\n","                          The masked alignment.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-tree\u001b[0m ARTIFACT       The unrooted\n","    \u001b[32mPhylogeny[Unrooted]\u001b[0m   phylogenetic tree.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-rooted-tree\u001b[0m ARTIFACT\n","    \u001b[32mPhylogeny[Rooted]\u001b[0m     The rooted phylogenetic\n","                          tree.         \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH       Output unspecified\n","                          results to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m     Display verbose output\n","                          to stdout and/or stderr\n","                          during execution of this\n","                          action. Or silence\n","                          output if execution is\n","                          successful (silence is\n","                          golden).\n","  \u001b[34m--examples\u001b[0m              Show usage examples and\n","                          exit.\n","  \u001b[34m--citations\u001b[0m             Show citations and\n","                          exit.\n","  \u001b[34m--help\u001b[0m                  Show this message and\n","                          exit.\n","\n","\u001b[33m                    There was a problem with the command:                     \u001b[0m\n","\u001b[31m\u001b[1m (1/1) Invalid value for '--i-sequences': 'rep-seqs-deblur.qza' is not a valid\n","  filepath\u001b[0m\n"]}]},{"cell_type":"markdown","metadata":{"id":"wxQVCxlg_6wL"},"source":["#alpha-rarefaction, view the plot on QIIME2 view "]},{"cell_type":"code","metadata":{"id":"p5QNgG8ol_lB","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1630551275019,"user_tz":-60,"elapsed":7210,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"fd062df4-da41-4536-e0af-832a8c7b636a"},"source":["!source activate qiime2-2020.11 && qiime diversity alpha-rarefaction \\\n","  --i-table table-deblur.qza \\\n","  --i-phylogeny rooted-tree.qza \\\n","  --p-max-depth 1000 \\\n","  --m-metadata-file AFmetadata.txt \\\n","  --o-visualization alpha-rarefaction.qzv"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime diversity alpha-rarefaction\u001b[0m \n","           [OPTIONS]\n","\n","  Generate interactive alpha rarefaction curves\n","  by computing rarefactions between `min_depth`\n","  and `max_depth`. The number of intermediate\n","  depths to compute is controlled by the `steps`\n","  parameter, with n `iterations` being computed\n","  at each rarefaction depth. If sample metadata\n","  is provided, samples may be grouped based on\n","  distinct values within a metadata column.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-table\u001b[0m ARTIFACT \u001b[32mFeatureTable[Frequency]\u001b[0m\n","                          Feature table to\n","                          compute rarefaction\n","                          curves from.  \u001b[35m[required]\u001b[0m\n","  \u001b[34m--i-phylogeny\u001b[0m ARTIFACT  Optional phylogeny for\n","    \u001b[32mPhylogeny[Rooted]\u001b[0m     phylogenetic metrics.\n","                                        \u001b[35m[optional]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m\u001b[4m--p-max-depth\u001b[0m INTEGER   The maximum rarefaction\n","    \u001b[32mRange(1, None)\u001b[0m        depth. Must be greater\n","                          than \u001b[4mmin-depth\u001b[0m.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m--p-metrics\u001b[0m TEXT... \u001b[32mChoices('mcintosh_e',\u001b[0m\n","    \u001b[32m'fisher_alpha', 'chao1', 'berger_parker_d',\u001b[0m\n","    \u001b[32m'singles', 'michaelis_menten_fit', 'pielou_e',\u001b[0m\n","    \u001b[32m'robbins', 'simpson_e', 'enspie', 'doubles',\u001b[0m\n","    \u001b[32m'lladser_pe', 'simpson', 'menhinick',\u001b[0m\n","    \u001b[32m'mcintosh_d', 'faith_pd', 'brillouin_d',\u001b[0m\n","    \u001b[32m'ace', 'dominance', 'heip_e', 'gini_index',\u001b[0m\n","    \u001b[32m'goods_coverage', 'observed_features',\u001b[0m\n","    \u001b[32m'margalef', 'shannon')\u001b[0m\n","                          The metrics to be\n","                          measured. By default\n","                          computes\n","                          observed_features,\n","                          shannon, and if\n","                          phylogeny is provided,\n","                          faith_pd.     \u001b[35m[optional]\u001b[0m\n","  \u001b[34m--m-metadata-file\u001b[0m METADATA...\n","    (multiple arguments   The sample metadata.\n","     will be merged)                    \u001b[35m[optional]\u001b[0m\n","  \u001b[34m--p-min-depth\u001b[0m INTEGER   The minimum rarefaction\n","    \u001b[32mRange(1, None)\u001b[0m        depth.      \u001b[35m[default: 1]\u001b[0m\n","  \u001b[34m--p-steps\u001b[0m INTEGER       The number of\n","    \u001b[32mRange(2, None)\u001b[0m        rarefaction depths to\n","                          include between\n","                          \u001b[4mmin-depth\u001b[0m and \u001b[4mmax-depth\u001b[0m.\n","                                     \u001b[35m[default: 10]\u001b[0m\n","  \u001b[34m--p-iterations\u001b[0m INTEGER  The number of rarefied\n","    \u001b[32mRange(1, None)\u001b[0m        feature tables to\n","                          compute at each step.\n","                                     \u001b[35m[default: 10]\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-visualization\u001b[0m VISUALIZATION\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH       Output unspecified\n","                          results to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m     Display verbose output\n","                          to stdout and/or stderr\n","                          during execution of this\n","                          action. Or silence\n","                          output if execution is\n","                          successful (silence is\n","                          golden).\n","  \u001b[34m--examples\u001b[0m              Show usage examples and\n","                          exit.\n","  \u001b[34m--citations\u001b[0m             Show citations and\n","                          exit.\n","  \u001b[34m--help\u001b[0m                  Show this message and\n","                          exit.\n","\n","\u001b[33m                    There was a problem with the command:                     \u001b[0m\n","\u001b[31m\u001b[1m (1/1) Invalid value for '--i-table': 'table-deblur.qza' is not a valid\n","  filepath\u001b[0m\n"]}]},{"cell_type":"markdown","metadata":{"id":"ibAgpC7r_9ll"},"source":["#Core metrics, sampling depth determined from rep-seqs-deblur.qzv"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"1q_hIlGRXdTn","executionInfo":{"status":"ok","timestamp":1630551281956,"user_tz":-60,"elapsed":6949,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"bcf6052d-eaa7-417d-c4e1-5eda3ccd22a7"},"source":["#Core metrics\n","!source activate qiime2-2020.11 && qiime diversity core-metrics-phylogenetic \\\n","  --i-phylogeny rooted-tree.qza \\\n","  --i-table table-deblur.qza \\\n","  --p-sampling-depth 1000 \\\n","  --m-metadata-file AFmetadata.txt \\\n","  --output-dir core-metrics-results"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime diversity core-metrics-phylogenetic\u001b[0m \n","           [OPTIONS]\n","\n","  Applies a collection of diversity metrics\n","  (both phylogenetic and non-phylogenetic) to a\n","  feature table.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-table\u001b[0m ARTIFACT \u001b[32mFeatureTable[Frequency]\u001b[0m\n","                          The feature table\n","                          containing the samples\n","                          over which diversity\n","                          metrics should be\n","                          computed.     \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--i-phylogeny\u001b[0m ARTIFACT  Phylogenetic tree\n","    \u001b[32mPhylogeny[Rooted]\u001b[0m     containing tip\n","                          identifiers that\n","                          correspond to the\n","                          feature identifiers in\n","                          the table. This tree can\n","                          contain tip ids that are\n","                          not present in the\n","                          table, but all feature\n","                          ids in the table must be\n","                          present in this tree.\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m\u001b[4m--p-sampling-depth\u001b[0m INTEGER\n","    \u001b[32mRange(1, None)\u001b[0m        The total frequency\n","                          that each sample should\n","                          be rarefied to prior to\n","                          computing diversity\n","                          metrics.      \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--m-metadata-file\u001b[0m METADATA...\n","    (multiple arguments   The sample metadata to\n","     will be merged)      use in the emperor\n","                          plots.        \u001b[35m[required]\u001b[0m\n","  \u001b[34m--p-n-jobs-or-threads\u001b[0m VALUE \u001b[32mInt % Range(1,\u001b[0m\n","    \u001b[32mNone) | Str % Choices('auto')\u001b[0m\n","                          [beta/beta-phylogenetic\n","                          methods only] - The\n","                          number of concurrent\n","                          jobs or CPU threads to\n","                          use in performing this\n","                          calculation. Individual\n","                          methods will create\n","                          jobs/threads as\n","                          implemented in\n","                          q2-diversity-lib\n","                          dependencies. May not\n","                          exceed the number of\n","                          available physical\n","                          cores. If\n","                          \u001b[4mn-jobs-or-threads\u001b[0m =\n","                          'auto', one thread/job\n","                          will be created for each\n","                          identified CPU core on\n","                          the host.   \u001b[35m[default: 1]\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-rarefied-table\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mFeatureTable[Frequency]\u001b[0m\n","                          The resulting rarefied\n","                          feature table.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-faith-pd-vector\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mSampleData[AlphaDiversity]\u001b[0m\n","                          Vector of Faith PD\n","                          values by sample.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-observed-features-vector\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mSampleData[AlphaDiversity]\u001b[0m\n","                          Vector of Observed\n","                          Features values by\n","                          sample.       \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-shannon-vector\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mSampleData[AlphaDiversity]\u001b[0m\n","                          Vector of Shannon\n","                          diversity values by\n","                          sample.       \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-evenness-vector\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mSampleData[AlphaDiversity]\u001b[0m\n","                          Vector of Pielou's\n","                          evenness values by\n","                          sample.       \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-unweighted-unifrac-distance-matrix\u001b[0m ARTIFACT\n","    \u001b[32mDistanceMatrix\u001b[0m        Matrix of unweighted\n","                          UniFrac distances\n","                          between pairs of\n","                          samples.      \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-weighted-unifrac-distance-matrix\u001b[0m ARTIFACT\n","    \u001b[32mDistanceMatrix\u001b[0m        Matrix of weighted\n","                          UniFrac distances\n","                          between pairs of\n","                          samples.      \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-jaccard-distance-matrix\u001b[0m ARTIFACT\n","    \u001b[32mDistanceMatrix\u001b[0m        Matrix of Jaccard\n","                          distances between pairs\n","                          of samples.   \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-bray-curtis-distance-matrix\u001b[0m ARTIFACT\n","    \u001b[32mDistanceMatrix\u001b[0m        Matrix of Bray-Curtis\n","                          distances between pairs\n","                          of samples.   \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-unweighted-unifrac-pcoa-results\u001b[0m ARTIFACT\n","    \u001b[32mPCoAResults\u001b[0m           PCoA matrix computed\n","                          from unweighted UniFrac\n","                          distances between\n","                          samples.      \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-weighted-unifrac-pcoa-results\u001b[0m ARTIFACT\n","    \u001b[32mPCoAResults\u001b[0m           PCoA matrix computed\n","                          from weighted UniFrac\n","                          distances between\n","                          samples.      \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-jaccard-pcoa-results\u001b[0m ARTIFACT\n","    \u001b[32mPCoAResults\u001b[0m           PCoA matrix computed\n","                          from Jaccard distances\n","                          between samples.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-bray-curtis-pcoa-results\u001b[0m ARTIFACT\n","    \u001b[32mPCoAResults\u001b[0m           PCoA matrix computed\n","                          from Bray-Curtis\n","                          distances between\n","                          samples.      \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-unweighted-unifrac-emperor\u001b[0m VISUALIZATION\n","                          Emperor plot of the\n","                          PCoA matrix computed\n","                          from unweighted UniFrac.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-weighted-unifrac-emperor\u001b[0m VISUALIZATION\n","                          Emperor plot of the\n","                          PCoA matrix computed\n","                          from weighted UniFrac.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-jaccard-emperor\u001b[0m VISUALIZATION\n","                          Emperor plot of the\n","                          PCoA matrix computed\n","                          from Jaccard. \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--o-bray-curtis-emperor\u001b[0m VISUALIZATION\n","                          Emperor plot of the\n","                          PCoA matrix computed\n","                          from Bray-Curtis.\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH       Output unspecified\n","                          results to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m     Display verbose output\n","                          to stdout and/or stderr\n","                          during execution of this\n","                          action. Or silence\n","                          output if execution is\n","                          successful (silence is\n","                          golden).\n","  \u001b[34m--examples\u001b[0m              Show usage examples and\n","                          exit.\n","  \u001b[34m--citations\u001b[0m             Show citations and\n","                          exit.\n","  \u001b[34m--help\u001b[0m                  Show this message and\n","                          exit.\n","\n","\u001b[33m                    There was a problem with the command:                     \u001b[0m\n","\u001b[31m\u001b[1m (1/1) Invalid value for '--i-table': 'table-deblur.qza' is not a valid\n","  filepath\u001b[0m\n"]}]},{"cell_type":"markdown","metadata":{"id":"watDCqQpAyKM"},"source":["#alpha statistics examples"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"SMvjOWkm7tdo","executionInfo":{"status":"ok","timestamp":1630551289910,"user_tz":-60,"elapsed":7959,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"efa5c3de-b94f-4d3b-85a9-c0d6eb7a5a49"},"source":["!source activate qiime2-2020.11 && qiime diversity alpha-group-significance \\\n","  --i-alpha-diversity core-metrics-results/faith_pd_vector.qza \\\n","  --m-metadata-file AFmetadata.txt \\\n","  --o-visualization core-metrics-results/faith-pd-group-significance.qzv\n","\n","!source activate qiime2-2020.11 && qiime diversity alpha-group-significance \\\n","  --i-alpha-diversity core-metrics-results/evenness_vector.qza \\\n","  --m-metadata-file AFmetadata.txt \\\n","  --o-visualization core-metrics-results/evenness-group-significance.qzv"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime diversity alpha-group-significance\u001b[0m \n","           [OPTIONS]\n","\n","  Visually and statistically compare groups of\n","  alpha diversity values.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-alpha-diversity\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mSampleData[AlphaDiversity]\u001b[0m\n","                       Vector of alpha diversity\n","                       values by sample.\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m\u001b[4m--m-metadata-file\u001b[0m METADATA...\n","    (multiple          The sample metadata.\n","     arguments will    \n","     be merged)                         \u001b[35m[required]\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-visualization\u001b[0m VISUALIZATION\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH    Output unspecified results\n","                       to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m  Display verbose output to\n","                       stdout and/or stderr during\n","                       execution of this action.\n","                       Or silence output if\n","                       execution is successful\n","                       (silence is golden).\n","  \u001b[34m--examples\u001b[0m           Show usage examples and\n","                       exit.\n","  \u001b[34m--citations\u001b[0m          Show citations and exit.\n","  \u001b[34m--help\u001b[0m               Show this message and\n","                       exit.\n","\n","\u001b[33m                  There were some problems with the command:                  \u001b[0m\n","\u001b[31m\u001b[1m (1/2) Invalid value for '--i-alpha-diversity': 'core-metrics-\n","  results/faith_pd_vector.qza' is not a valid filepath\u001b[0m\n","\u001b[31m\u001b[1m (2/2) Invalid value for '--o-visualization': Directory 'core-metrics-results'\n","  does not exist, cannot save 'faith-pd-group-significance.qzv' into it.\u001b[0m\n","Usage: \u001b[34mqiime diversity alpha-group-significance\u001b[0m \n","           [OPTIONS]\n","\n","  Visually and statistically compare groups of\n","  alpha diversity values.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-alpha-diversity\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mSampleData[AlphaDiversity]\u001b[0m\n","                       Vector of alpha diversity\n","                       values by sample.\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m\u001b[4m--m-metadata-file\u001b[0m METADATA...\n","    (multiple          The sample metadata.\n","     arguments will    \n","     be merged)                         \u001b[35m[required]\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-visualization\u001b[0m VISUALIZATION\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH    Output unspecified results\n","                       to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m  Display verbose output to\n","                       stdout and/or stderr during\n","                       execution of this action.\n","                       Or silence output if\n","                       execution is successful\n","                       (silence is golden).\n","  \u001b[34m--examples\u001b[0m           Show usage examples and\n","                       exit.\n","  \u001b[34m--citations\u001b[0m          Show citations and exit.\n","  \u001b[34m--help\u001b[0m               Show this message and\n","                       exit.\n","\n","\u001b[33m                  There were some problems with the command:                  \u001b[0m\n","\u001b[31m\u001b[1m (1/2) Invalid value for '--i-alpha-diversity': 'core-metrics-\n","  results/evenness_vector.qza' is not a valid filepath\u001b[0m\n","\u001b[31m\u001b[1m (2/2) Invalid value for '--o-visualization': Directory 'core-metrics-results'\n","  does not exist, cannot save 'evenness-group-significance.qzv' into it.\u001b[0m\n"]}]},{"cell_type":"markdown","metadata":{"id":"1bx273_lA3yj"},"source":["#beta statistics examples"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"HpLf9PA_77as","executionInfo":{"status":"ok","timestamp":1630551293849,"user_tz":-60,"elapsed":3952,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"70c07916-27ba-4874-a273-9fc7e903bc0f"},"source":["!source activate qiime2-2020.11 && qiime diversity beta-group-significance \\\n","  --i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \\\n","  --m-metadata-file AFmetadata.txt \\\n","  --m-metadata-column group \\\n","  --o-visualization core-metrics-results/unweighted-unifrac-Ethnicity-significance.qzv \\\n","  --p-pairwise"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime diversity beta-group-significance\u001b[0m \n","           [OPTIONS]\n","\n","  Determine whether groups of samples are\n","  significantly different from one another using\n","  a permutation-based statistical test.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-distance-matrix\u001b[0m ARTIFACT\n","    \u001b[32mDistanceMatrix\u001b[0m     Matrix of distances\n","                       between pairs of samples.\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m\u001b[4m--m-metadata-file\u001b[0m METADATA\n","  \u001b[34m\u001b[4m--m-metadata-column\u001b[0m COLUMN  \u001b[32m\u001b[0m\n","    \u001b[32mMetadataColumn[Categorical]\u001b[0m\n","                       Categorical sample\n","                       metadata column. \u001b[35m[required]\u001b[0m\n","  \u001b[34m--p-method\u001b[0m TEXT \u001b[32mChoices('permanova',\u001b[0m\n","    \u001b[32m'anosim', 'permdisp')\u001b[0m\n","                       The group significance\n","                       test to be applied.\n","                            \u001b[35m[default: 'permanova']\u001b[0m\n","  \u001b[34m--p-pairwise\u001b[0m / \u001b[34m--p-no-pairwise\u001b[0m\n","                       Perform pairwise tests\n","                       between all pairs of groups\n","                       in addition to the test\n","                       across all groups. This can\n","                       be very slow if there are a\n","                       lot of groups in the\n","                       metadata column.\n","                                  \u001b[35m[default: False]\u001b[0m\n","  \u001b[34m--p-permutations\u001b[0m INTEGER\n","                       The number of permutations\n","                       to be run when computing\n","                       p-values.    \u001b[35m[default: 999]\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-visualization\u001b[0m VISUALIZATION\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH    Output unspecified results\n","                       to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m  Display verbose output to\n","                       stdout and/or stderr during\n","                       execution of this action.\n","                       Or silence output if\n","                       execution is successful\n","                       (silence is golden).\n","  \u001b[34m--examples\u001b[0m           Show usage examples and\n","                       exit.\n","  \u001b[34m--citations\u001b[0m          Show citations and exit.\n","  \u001b[34m--help\u001b[0m               Show this message and\n","                       exit.\n","\n","\u001b[33m                  There were some problems with the command:                  \u001b[0m\n","\u001b[31m\u001b[1m (1/3) Invalid value for '--i-distance-matrix': 'core-metrics-\n","  results/unweighted_unifrac_distance_matrix.qza' is not a valid filepath\u001b[0m\n","\u001b[31m\u001b[1m (2/3) Invalid value for '--m-metadata-file': There was an issue with\n","  retrieving column 'group' from the metadata.\u001b[0m\n","\u001b[31m\u001b[1m (3/3) Invalid value for '--o-visualization': Directory 'core-metrics-results'\n","  does not exist, cannot save 'unweighted-unifrac-Ethnicity-significance.qzv'\n","  into it.\u001b[0m\n"]}]},{"cell_type":"markdown","metadata":{"id":"9oxGo5H3IOGs"},"source":["#Taxonomy(needs checking)"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"0JEQm-Kx-GYD","executionInfo":{"status":"ok","timestamp":1630551296943,"user_tz":-60,"elapsed":3108,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"96467185-6754-427f-f636-f062ab7c61d5"},"source":["!wget \\\n","  -O \"gg-13-8-99-nb-classifier.qza\" \\\n","  \"https://data.qiime2.org/2020.11/common/gg-13-8-99-nb-classifier.qza\""],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["--2021-09-02 02:54:53--  https://data.qiime2.org/2020.11/common/gg-13-8-99-nb-classifier.qza\n","Resolving data.qiime2.org (data.qiime2.org)... 54.200.1.12\n","Connecting to data.qiime2.org (data.qiime2.org)|54.200.1.12|:443... connected.\n","HTTP request sent, awaiting response... 302 FOUND\n","Location: https://s3-us-west-2.amazonaws.com/qiime2-data/2020.11/common/gg-13-8-99-nb-classifier.qza [following]\n","--2021-09-02 02:54:53--  https://s3-us-west-2.amazonaws.com/qiime2-data/2020.11/common/gg-13-8-99-nb-classifier.qza\n","Resolving s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)... 52.218.224.56\n","Connecting to s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)|52.218.224.56|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 104510980 (100M) [binary/octet-stream]\n","Saving to: ‘gg-13-8-99-nb-classifier.qza’\n","\n","gg-13-8-99-nb-class 100%[===================>]  99.67M  33.8MB/s    in 2.9s    \n","\n","2021-09-02 02:54:56 (33.8 MB/s) - ‘gg-13-8-99-nb-classifier.qza’ saved [104510980/104510980]\n","\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"VkWA8v-fA5TH","executionInfo":{"status":"ok","timestamp":1630551300857,"user_tz":-60,"elapsed":3923,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"4c075453-ea3f-4a10-88ea-cdc039b2f31c"},"source":["!source activate qiime2-2020.11 && qiime feature-classifier classify-sklearn \\\n","  --i-reads rep-seqs-deblur.qza \\\n","  --i-classifier /content/gg-13-8-99-nb-classifier.qza \\\n","  --o-classification taxonomy.qza"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime feature-classifier classify-sklearn\u001b[0m \n","           [OPTIONS]\n","\n","  Classify reads by taxon using a fitted\n","  classifier.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-reads\u001b[0m ARTIFACT \u001b[32mFeatureData[Sequence]\u001b[0m\n","                         The feature data to be\n","                         classified.    \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--i-classifier\u001b[0m ARTIFACT\n","    \u001b[32mTaxonomicClassifier\u001b[0m  The taxonomic classifier\n","                         for classifying the\n","                         reads.         \u001b[35m[required]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m--p-reads-per-batch\u001b[0m INTEGER\n","    \u001b[32mRange(0, None)\u001b[0m       Number of reads to\n","                         process in each batch. If\n","                         \"auto\", this parameter is\n","                         autoscaled to min( number\n","                         of query sequences /\n","                         \u001b[4mn-jobs\u001b[0m, 20000).\n","                                      \u001b[35m[default: 0]\u001b[0m\n","  \u001b[34m--p-n-jobs\u001b[0m INTEGER     The maximum number of\n","                         concurrently worker\n","                         processes. If -1 all CPUs\n","                         are used. If 1 is given,\n","                         no parallel computing\n","                         code is used at all,\n","                         which is useful for\n","                         debugging. For \u001b[4mn-jobs\u001b[0m\n","                         below -1, (n_cpus + 1 +\n","                         \u001b[4mn-jobs\u001b[0m) are used. Thus\n","                         for \u001b[4mn-jobs\u001b[0m = -2, all CPUs\n","                         but one are used.\n","                                      \u001b[35m[default: 1]\u001b[0m\n","  \u001b[34m--p-pre-dispatch\u001b[0m TEXT  \"all\" or expression, as\n","                         in \"3*n_jobs\". The number\n","                         of batches (of tasks) to\n","                         be pre-dispatched.\n","                             \u001b[35m[default: '2*n_jobs']\u001b[0m\n","  \u001b[34m--p-confidence\u001b[0m VALUE \u001b[32mFloat % Range(0, 1,\u001b[0m\n","    \u001b[32minclusive_end=True) | Str % Choices('disable')\u001b[0m\n","                         Confidence threshold for\n","                         limiting taxonomic depth.\n","                         Set to \"disable\" to\n","                         disable confidence\n","                         calculation, or 0 to\n","                         calculate confidence but\n","                         not apply it to limit the\n","                         taxonomic depth of the\n","                         assignments.\n","                                    \u001b[35m[default: 0.7]\u001b[0m\n","  \u001b[34m--p-read-orientation\u001b[0m TEXT \u001b[32mChoices('same',\u001b[0m\n","    \u001b[32m'reverse-complement', 'auto')\u001b[0m\n","                         Direction of reads with\n","                         respect to reference\n","                         sequences. same will\n","                         cause reads to be\n","                         classified unchanged;\n","                         reverse-complement will\n","                         cause reads to be\n","                         reversed and complemented\n","                         prior to classification.\n","                         \"auto\" will autodetect\n","                         orientation based on the\n","                         confidence estimates for\n","                         the first 100 reads.\n","                                 \u001b[35m[default: 'auto']\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-classification\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mFeatureData[Taxonomy]\u001b[0m\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH      Output unspecified\n","                         results to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m    Display verbose output\n","                         to stdout and/or stderr\n","                         during execution of this\n","                         action. Or silence output\n","                         if execution is\n","                         successful (silence is\n","                         golden).\n","  \u001b[34m--examples\u001b[0m             Show usage examples and\n","                         exit.\n","  \u001b[34m--citations\u001b[0m            Show citations and exit.\n","  \u001b[34m--help\u001b[0m                 Show this message and\n","                         exit.\n","\n","\u001b[33m                  There were some problems with the command:                  \u001b[0m\n","\u001b[31m\u001b[1m (1/2) Invalid value for '--i-reads': 'rep-seqs-deblur.qza' is not a valid\n","  filepath\u001b[0m\n","\u001b[31m\u001b[1m (2/2) Invalid value for '--i-classifier': '/content/gg-13-8-99-nb-\n","  classifier.qza' is not a valid filepath\u001b[0m\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"E9_LmA66ZHsD","executionInfo":{"status":"ok","timestamp":1630551308818,"user_tz":-60,"elapsed":7972,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"71b0ce54-0e0d-40a4-941f-d179b6d69457"},"source":["!source activate qiime2-2020.11 && qiime metadata tabulate \\\n","  --m-input-file taxonomy.qza \\\n","  --o-visualization taxonomy.qzv"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[32mSaved Visualization to: taxonomy.qzv\u001b[0m\n"]}]},{"cell_type":"markdown","metadata":{"id":"nwuOe8awXWwc"},"source":["#?min frequency is based on alpha"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"tm9Z7ddWaew1","executionInfo":{"status":"ok","timestamp":1630551312897,"user_tz":-60,"elapsed":4090,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"5961b764-619a-4371-e8e5-bad06fe49ccb"},"source":["!source activate qiime2-2020.11 && qiime feature-table filter-samples \\\n","  --i-table table-deblur.qza \\\n","  --p-min-frequency 1000 \\\n","  --o-filtered-table table_taxa.qza\n"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime feature-table filter-samples\u001b[0m \n","           [OPTIONS]\n","\n","  Filter samples from table based on frequency\n","  and/or metadata. Any features with a frequency\n","  of zero after sample filtering will also be\n","  removed. See the filtering tutorial on\n","  https://docs.qiime2.org for additional\n","  details.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-table\u001b[0m ARTIFACT \u001b[32mFeatureTable[Frequency¹ |\u001b[0m\n","    \u001b[32mRelativeFrequency² | PresenceAbsence³ |\u001b[0m\n","    \u001b[32mComposition⁴]\u001b[0m      The feature table from\n","                       which samples should be\n","                       filtered.        \u001b[35m[required]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m--p-min-frequency\u001b[0m INTEGER\n","                       The minimum total\n","                       frequency that a sample\n","                       must have to be retained.\n","                                      \u001b[35m[default: 0]\u001b[0m\n","  \u001b[34m--p-max-frequency\u001b[0m INTEGER\n","                       The maximum total\n","                       frequency that a sample can\n","                       have to be retained. If no\n","                       value is provided this will\n","                       default to infinity (i.e.,\n","                       no maximum frequency filter\n","                       will be applied).\n","                                        \u001b[35m[optional]\u001b[0m\n","  \u001b[34m--p-min-features\u001b[0m INTEGER\n","                       The minimum number of\n","                       features that a sample must\n","                       have to be retained.\n","                                      \u001b[35m[default: 0]\u001b[0m\n","  \u001b[34m--p-max-features\u001b[0m INTEGER\n","                       The maximum number of\n","                       features that a sample can\n","                       have to be retained. If no\n","                       value is provided this will\n","                       default to infinity (i.e.,\n","                       no maximum feature filter\n","                       will be applied).\n","                                        \u001b[35m[optional]\u001b[0m\n","  \u001b[34m--m-metadata-file\u001b[0m METADATA...\n","    (multiple          Sample metadata used with\n","     arguments will    `where` parameter when\n","     be merged)        selecting samples to\n","                       retain, or with\n","                       `\u001b[4mexclude-ids\u001b[0m` when\n","                       selecting samples to\n","                       discard.         \u001b[35m[optional]\u001b[0m\n","  \u001b[34m--p-where\u001b[0m TEXT       SQLite WHERE clause\n","                       specifying sample metadata\n","                       criteria that must be met\n","                       to be included in the\n","                       filtered feature table. If\n","                       not provided, all samples\n","                       in `metadata` that are also\n","                       in the feature table will\n","                       be retained.     \u001b[35m[optional]\u001b[0m\n","  \u001b[34m--p-exclude-ids\u001b[0m / \u001b[34m--p-no-exclude-ids\u001b[0m\n","                       If true, the samples\n","                       selected by `metadata` or\n","                       `where` parameters will be\n","                       excluded from the filtered\n","                       table instead of being\n","                       retained.  \u001b[35m[default: False]\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-filtered-table\u001b[0m ARTIFACT \u001b[32m\u001b[0m\n","    \u001b[32mFeatureTable[Frequency¹ | RelativeFrequency² |\u001b[0m\n","    \u001b[32mPresenceAbsence³ | Composition⁴]\u001b[0m\n","                       The resulting feature\n","                       table filtered by sample.\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH    Output unspecified results\n","                       to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m  Display verbose output to\n","                       stdout and/or stderr during\n","                       execution of this action.\n","                       Or silence output if\n","                       execution is successful\n","                       (silence is golden).\n","  \u001b[34m--examples\u001b[0m           Show usage examples and\n","                       exit.\n","  \u001b[34m--citations\u001b[0m          Show citations and exit.\n","  \u001b[34m--help\u001b[0m               Show this message and\n","                       exit.\n","\n","\u001b[33m                    There was a problem with the command:                     \u001b[0m\n","\u001b[31m\u001b[1m (1/1) Invalid value for '--i-table': 'table-deblur.qza' is not a valid\n","  filepath\u001b[0m\n"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"JeVsw1OZckmc","executionInfo":{"status":"ok","timestamp":1630551319986,"user_tz":-60,"elapsed":7101,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"5ffed628-fea1-45b9-d8bb-faffc87b702d"},"source":["!source activate qiime2-2020.11 && qiime taxa barplot \\\n","  --i-table table_taxa.qza \\\n","  --i-taxonomy taxonomy.qza \\\n","  --m-metadata-file AFmetadata.txt \\\n","  --o-visualization taxa_barplot.qzv"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Usage: \u001b[34mqiime taxa barplot\u001b[0m [OPTIONS]\n","\n","  This visualizer produces an interactive\n","  barplot visualization of taxonomies.\n","  Interactive features include multi-level\n","  sorting, plot recoloring, sample relabeling,\n","  and SVG figure export.\n","\n","\u001b[1mInputs\u001b[0m:\n","  \u001b[34m\u001b[4m--i-table\u001b[0m ARTIFACT \u001b[32mFeatureTable[Frequency]\u001b[0m\n","                         Feature table to\n","                         visualize at various\n","                         taxonomic levels.\n","                                        \u001b[35m[required]\u001b[0m\n","  \u001b[34m\u001b[4m--i-taxonomy\u001b[0m ARTIFACT \u001b[32mFeatureData[Taxonomy]\u001b[0m\n","                         Taxonomic annotations\n","                         for features in the\n","                         provided feature table.\n","                         All features in the\n","                         feature table must have a\n","                         corresponding taxonomic\n","                         annotation. Taxonomic\n","                         annotations that are not\n","                         present in the feature\n","                         table will be ignored.\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mParameters\u001b[0m:\n","  \u001b[34m\u001b[4m--m-metadata-file\u001b[0m METADATA...\n","    (multiple            The sample metadata.\n","     arguments will be   \n","     merged)                            \u001b[35m[required]\u001b[0m\n","\u001b[1mOutputs\u001b[0m:\n","  \u001b[34m\u001b[4m--o-visualization\u001b[0m VISUALIZATION\n","                                        \u001b[35m[required]\u001b[0m\n","\u001b[1mMiscellaneous\u001b[0m:\n","  \u001b[34m--output-dir\u001b[0m PATH      Output unspecified\n","                         results to a directory\n","  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m    Display verbose output\n","                         to stdout and/or stderr\n","                         during execution of this\n","                         action. Or silence output\n","                         if execution is\n","                         successful (silence is\n","                         golden).\n","  \u001b[34m--examples\u001b[0m             Show usage examples and\n","                         exit.\n","  \u001b[34m--citations\u001b[0m            Show citations and exit.\n","  \u001b[34m--help\u001b[0m                 Show this message and\n","                         exit.\n","\n","\u001b[33m                    There was a problem with the command:                     \u001b[0m\n","\u001b[31m\u001b[1m (1/1) Invalid value for '--i-table': 'table_taxa.qza' is not a valid filepath\u001b[0m\n"]}]},{"cell_type":"markdown","metadata":{"id":"CUBJpsh4MLgQ"},"source":["#**To save the outputs (qzv and qza files)** "]},{"cell_type":"code","metadata":{"id":"kCQn2Co5DQx4","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1630551319987,"user_tz":-60,"elapsed":19,"user":{"displayName":"Andrej Hric","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhV9-R5teGHMkMyHv9sMdL2S1qrr3C6nV3BsvBe=s64","userId":"05537033133798647470"}},"outputId":"3c703d5d-c844-4cb2-eaa9-c78e439877f4"},"source":["!cp /content/sample_data/core-metrics-results/*.qz* /content/drive/MyDrive/ECgroup\n","!cp /content/sample_data/*.qz* /content/drive/Mydrive/ECgroup"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["cp: cannot stat '/content/sample_data/core-metrics-results/*.qz*': No such file or directory\n","cp: target '/content/drive/Mydrive/ECgroup' is not a directory\n"]}]}]}